---
title: "dOTS Analysis"
author: "Nicholas Popp"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
    #pdf_document:
        #latex_engine: xelatex
        # toc: TRUE
        # toc_depth: 4
        #number_sections: TRUE
        # fig_caption: TRUE
    html_document
---

```{r setup, include = TRUE}
knitr::opts_knit$set(echo = TRUE, root.dir = '..')
```

# Introduction

This is a document for analyzing all the data generated for the dOTS project and manuscript. This file takes in multiple compiled .csv files that have each experiment laid out and plots data for figures. Data is generated from Illumina sequencing files by a custom script (see **Seq_analysis** folder) to output individual .csv files for each genomic locus. These have been manually compiled into .csv files which are stored in the **Data** folder.

```{r libaries}
## install and load libraries

## knitr for PDF creation
if (!require(knitr)) install.packages('knitr')
library(knitr)

## tidyverse for ggplot, dplyr, data manipulation
if (!require(tidyverse)) install.packages('tidyverse')
library(tidyverse)

## cowplot for plot organization
if (!require(cowplot)) install.packages('cowplot')
library(cowplot)

## kableExtra for table creation
if (!require(kableExtra)) install.packages('kableExtra')
library(kableExtra)

## ggbeeswarm for jiggering plots better than geom_jigger
## does not work with NAs
if (!require(ggbeeswarm)) install.packages('ggbeeswarm')
library(ggbeeswarm)

## patchwork for aligning plots
if (!require(patchwork)) install.packages('patchwork')
library(patchwork)

## magick for drawing images
if (!require(magick)) install.packages('magick')
library(magick)

## openxlsx for making source data file
if (!require(openxlsx)) install.packages('openxlsx')
library(openxlsx)

## Create directories for figures, tables, etc.
dir.create(path = "./Stat_analysis/Main_figures")
dir.create(path = "./Stat_analysis/Supp_figures")
dir.create(path = "./Stat_analysis/Supp_tables")

## set up color palette
pal <- c("#CA0020", "#92C5DE", "#F4A582", "#0571B0", "#999999")

## set up ggplot to look pretty for cowplot grid
ggplot <- function(...) {
  ggplot2::ggplot(...) + 
    theme_cowplot() +
    ## hide gridlines
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          ## change legend details
          legend.position = "top",
          legend.justification = "center",
          legend.key = element_rect(fill = "white", size = 0.7),
          legend.background = element_rect(fill = "NA"),
          legend.title = element_blank(),
          ## change text sizes
          axis.title = element_text(size = 15, color = "black"),
          axis.title.x = element_blank(),
          axis.text = element_text(size = 13, color = "black"),
          legend.text = element_text(size = 13, color = "black"),
          ## adjust plot title location
          plot.title = element_text(hjust = 0.5)) +
    ## allow points to be plotted on axes
    coord_cartesian(clip = "off")
}

## blank ggplot with labels for alignment
label_plot <- function(...){
  ggplot(...) +
  theme(axis.line = element_blank(),
        axis.ticks = element_blank(),
        axis.text.y = element_blank(),
        axis.title = element_blank(),
        axis.text.x = element_text(hjust = 0),
        plot.title = element_text(hjust = 0.5, size = 36)) 
}

## function to format x axis label
addline_format <- function(x){
  y <- gsub("^\\s", "", x)
  y <- gsub("\\s", "\n\n", y)
  paste0("\n", y)
}

## function to grab guides and create labels for plot axes
label_grab <- function(x){
  grab <- x$guide
  new_label <- paste0("\nCas9\n\n", grab, "\n\ndRNA")
  return(new_label)
}
  
## function to calculate ratios
ratio_calc <- function(x){
  ## filter out low reads and negative controls
  temp_ratio <- filter(x, Reads_passing >= 1000 &
                       X_lab != "- - -" &
                       X_lab != "- - - -" &
                       X_lab != "- - - - -") %>%
  ## divide for future join
  separate(col = Site, into = c("guide", "location"), sep = " (?=[^ ]+$)")

  ## split into two temporary  files
  ratio_on <- filter(temp_ratio, location == "ON")
  ratio_off <- filter(temp_ratio, location != "ON") %>%
    ## ensure nonzero to prevent inifite ratios
    mutate(Indel_stringent = replace(Indel_stringent,
                                Indel_stringent == 0,
                                0.01))
  
  ## join based on guide, plot, calculate ratio, plot
  ratio_plots <- full_join(ratio_on, ratio_off,
                           by = c("Experiment", "Name", "X_lab", "guide"),
                           suffix = c("_ON", "_OT")) %>%
    mutate(ratio = Indel_stringent_ON / Indel_stringent_OT) # calculates ratio

return(ratio_plots)
}

## calculate ratios for hifi 
ratio_calc_hifi <- function(x){
  ## filter out low read counts and negative controls
  temp_ratio <- filter(x, Reads_passing >= 1000 &
                       X_lab != "- - -" &
                       X_lab != "- - - -" &
                       X_lab != "- - - - -") %>% 
  ## divide
  separate(col = Site, into = c("guide", "location"), sep = " (?=[^ ]+$)")

  ## split into two temporary files
  ratio_on <- filter(temp_ratio, location == "ON")
  ratio_off <- filter(temp_ratio, location != "ON") %>%
    ## make sure all nonzero so no infinite ratios
    mutate(Indel_stringent = replace(Indel_stringent,
                                Indel_stringent == 0,
                                0.01))
  
  ## join based on guide and calculate ratio
  ratio_plots <- full_join(ratio_on, ratio_off,
                           by = c("Experiment", "Name",
                                  "X_lab", "guide",
                                  "X_order", "Cas9"),
                           suffix = c("_ON", "_OT")) %>%
    mutate(ratio = Indel_stringent_ON / Indel_stringent_OT)

return(ratio_plots)
}

## longer version for multi-dRNAs
ratio_calc_longer <- function(x){
  ## filter out less than 1000 reads and neg controls
  temp_ratio <- filter(x, Reads_passing >= 1000 &
                       X_lab != "- - -") %>% 
  ## divide site 
  separate(col = Site, into = c("guide", "location"), sep = " (?=[^ ]+$)")

  ## split into two temporary  files
  ratio_on <- filter(temp_ratio, location == "ON")
  ratio_off <- filter(temp_ratio, location != "ON") %>%
    ## make sure all nonzero so no infinite ratios
    mutate(Indel_stringent = replace(Indel_stringent,
                                     Indel_stringent == 0,
                                     0.01))
  
  ## join based on guide, calculate ratio
  ratio_plots <- full_join(ratio_on, ratio_off,
                           by = c("Experiment", "Name", "X_lab", "guide"),
                           suffix = c("_ON", "_OT")) %>%
    mutate(ratio = Indel_stringent_ON / Indel_stringent_OT) 
  
return(ratio_plots)
}

## add max value to plots and divide evenly for nice plotting
round_max <- function(x){
  ## find max plotted value
  if ("Indel_stringent" %in% colnames(x)){
    maxval <- max(x$Indel_stringent) * 1.1
  } else {
    maxval <- max(x$ratio) * 1.1
  }
  
  ## round the value to no significant digits
  roundmax <- round(maxval, digits = 0)
  
  ## round values to nearest 5 if less than 25
  if (roundmax <= 25){
    if (roundmax %% 5 != 0){
      roundmax = roundmax + 5 - (roundmax %% 5)
    }
    ## round values to nearest 10 if between 25 and 100
  } else if (roundmax > 25 & roundmax <= 100){
    if (roundmax %% 10 != 0){
      roundmax = roundmax + 10 - (roundmax %% 10)
    }
    ## round values to nearest 50 if between 100 and 250
  } else if (roundmax > 100 & roundmax <= 250){
    if (roundmax %% 50 != 0){
      roundmax = roundmax + 50 - (roundmax %% 50)
    }
    ## round values to nearest 100 if greater than 250
  } else if (roundmax >= 250){
    if (roundmax %% 100 != 0){
      roundmax = roundmax + 100 - (roundmax %% 100)
    }
  }
  
  return(roundmax)
}

## grouped split function for generating source data file
named_group_split <- function(.tbl, ...) {
  grouped <- group_by(.tbl, ...)
  names <- rlang::eval_bare(rlang::expr(paste(!!!group_keys(grouped), sep = " / ")))

  grouped %>% 
    group_split(keep = FALSE) %>% 
    rlang::set_names(names)
}

## function to create source data file for a figure with indel data
indel_source <- function(.tbl, ...){
  group_by(.tbl, Experiment) %>%
  mutate(target2 = if_else(str_detect(target, "OT") == TRUE,
                          "OT", "ON")) %>%
  ungroup() %>%
  select(Panel, Name, guide, X_lab, Indel_stringent, target, target2) %>%
  pivot_wider(names_from = target2,
              values_from = c(Indel_stringent, target)) %>%
  arrange(Panel, Name) %>%
  select(Panel, Name, guide, target_OT, X_lab, Indel_stringent_ON, Indel_stringent_OT) %>%
  rename("Indels ON" = Indel_stringent_ON,
         "Indels OT" = Indel_stringent_OT,
         "Off-target" = target_OT,
         "X label" = X_lab,
         "Guide" = guide)
}

## function to create source data file for a figure with ratio data
ratio_source <- function(.tbl, ...){
  arrange(.tbl, Panel, Name) %>%
  select(Panel, Name, guide, location_OT, X_lab, ratio) %>%
  rename("Off-target" = location_OT,
         "X label" = X_lab,
         "Guide" = guide,
         "Ratio" = ratio)
}

```

# dRNA screens for all tested sites

This section analyzes first pass dRNA screening experiments for 19 on-off target pairs to detect reduced off-target editing with co-incubation of a dRNA.

```{r dRNA screens}
## read dRNA screen .csv file into memory
dRNA_screens <- read_csv("Data/dRNA_screens.csv") %>%
  ## grab guide name
  mutate(guide = word(Site, start = 1),
         Site = gsub("g", " sgRNA", Site),
         guide = gsub("g", " sgRNA", guide),
         target = word(Site, start = -1)) %>%
  ## filter out m4d1 from all samples
  filter(X_lab != "+ + m4d1")

## ===========================================================================================

## generate dRNA screen plots as list
clean_screens <- dRNA_screens %>%
  filter(Reads_passing >= 1000 &
         Experiment != 3 &
         ## filter out samples that failed QC
         !grepl("^R0[0-9]_d[5-6]", Sample) &
         !grepl("^NP_JR_062517_R01_d6", Sample) &
         !grepl("^NP_JR_062517_R04_d6", Sample) &
         !grepl("^VEGFAg3_OT18d1", Name) &
         !grepl("^VEGFAg2_OT2d1", Name) &
         !grepl("^VEGFAg2_OT2d5", Name) &
         ## filter out experiments used in other plots
         ## or that failed QC
         Experiment != 24 & 
         Experiment != 13 & 
         Experiment != 14 &
         Experiment != 11) %>%
  group_by(Experiment) %>%
  arrange(Site) %>%
  ungroup() %>%
  mutate(Experiment = factor(Experiment, unique(Experiment)),
         Experiment = fct_relevel(Experiment, "19", after = 10)) %>%
  arrange(Experiment)

screen_plots <- clean_screens %>%
  group_by(Experiment) %>%
  mutate(tag = letters[Experiment]) %>%
  nest() %>%
  ## make indel plot
  mutate(plot = map2(data, Experiment,
                     ~ggplot(data = .x) +
                       aes(fct_inorder(X_lab),
                           Indel_stringent, fill = Site) +
                       geom_beeswarm(cex = 2, size = 2.5,
                                     pch = 21, alpha = 0.7) +
                       stat_summary(data = .x, fun = "mean",
                                    geom = "point", pch = "_",
                                    size = 8) +
                       scale_x_discrete(labels = addline_format(
                         unique(.$X_lab))) +
                       scale_y_continuous(expand = c(0, 0),
                                          limits = c(0, round_max(.)),
                                          breaks = seq(0, round_max(.),
                                                       by = if_else(round_max(.) <= 10, round_max(.) / 5,
                                                                    if_else(round_max(.) >= 30, 10, 5)))) + 
                       scale_fill_manual(values = pal,
                                         breaks = unique(.$Site),
                                         labels = c("on-target",
                                                    "off-target")) +
                       labs(y = "Indel frequency (%)\n",
                            title = unique(.$Site)[2])),
         ## add subtitles and x-axis labels
         label_plots = map2(data, Experiment,
                           ~label_plot(data = .x) +
                             aes(Date,
                                 Indel_stringent) +
                             geom_point(color = "NA") +
                             labs(title = unique(.$tag)) +
                             scale_x_discrete(labels = label_grab(.)))) %>%
  ## join to plot side by side
  pivot_longer(cols = c("label_plots", "plot"),
               names_to = "plottype",
               values_to = "plot")

## plot figures together, with labels
fig_s3 <- wrap_plots(screen_plots$plot,
                     ncol = 8,
                     widths = c(0.25, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, "pt"))

## save figure S3 - all dRNA screens
ggsave("Stat_analysis/Supp_figures/FigS3_dRNA_screen.pdf", plot = fig_s3,
       height = 30, width = 24, units = "in",
       limitsize = FALSE)

## make Fig. S3 data frame for source data file
fig_s3_source <- clean_screens %>%
  group_by(Experiment) %>%
  mutate(Panel = letters[Experiment]) %>%
  ungroup() %>%
  indel_source()

## split by panel
fig_s3_list <- fig_s3_source %>%
  named_group_split(Panel)

## change names to figure panels
names(fig_s3_list) <- paste0("FigS3", names(fig_s3_list))

## create and add to source list
source_data_list <- fig_s3_list

# ===========================================================================================

## prepare FANCFg2 OT1 dRNA screen for Figure 1b
fancf_screen <- filter(dRNA_screens, Experiment == 3 &
                       X_lab != "+ + 4") %>%
  mutate(X_lab = gsub("5", "4", X_lab))

## calculate ratios for FANCF
fancf_screen_ratio <- ratio_calc(fancf_screen)

## create labels for Figure S1
fig_s1_label <- label_plot(data = fancf_screen,
                            aes(x = as_factor(Experiment),
                                y = Indel_stringent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = label_grab(fancf_screen))

## label S1b
label_s1b <- fig_s1_label + 
  labs(title = "b")

## label S1c
label_s1c <- fig_s1_label +
  labs(title = "c")

## Figure S1b - indels
fig_s1b1 <- ggplot(data = fancf_screen,
                   aes(x = fct_inorder(X_lab),
                       y = Indel_stringent,
                       fill = Site)) +
  geom_beeswarm(cex = 2, size = 2.5,
                pch = 21, alpha = 0.7) +
  stat_summary(data = fancf_screen, fun = "mean",
               geom = "point", pch = "_", size = 8) + 
  scale_x_discrete(labels = addline_format(unique(fancf_screen$X_lab))) +
  scale_fill_manual(values = pal,
                     breaks = unique(fancf_screen$Site),
                     labels = c("on-target", "off-target")) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(fancf_screen)),
                     breaks = seq(0, round_max(fancf_screen),
                                  by = if_else(round_max(fancf_screen) <= 10, round_max(fancf_screen) / 5,
                                                                if_else(round_max(fancf_screen) >= 30, 10, 5)))) +
  labs(y = "Indel frequency (%)\n",
       title = unique(fancf_screen$Site)[2])

## Figure S1b - ratios
fig_s1b2 <- ggplot(data = fancf_screen_ratio,
                   aes(x = fct_inorder(X_lab), y = ratio)) +
  geom_beeswarm(cex = 2, size = 2.5,
                dodge.width = 0.5, pch = 21, alpha = 0.7, fill = pal[5]) +
  stat_summary(data = fancf_screen_ratio, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(fancf_screen_ratio$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(fancf_screen_ratio)),
                     breaks = seq(0, round_max(fancf_screen_ratio),
                                  by = if_else(round_max(fancf_screen_ratio) <= 10,
                                               round_max(fancf_screen_ratio) / 5,
                                               if_else(round_max(fancf_screen_ratio) >= 80, 20, 5)))) +
  labs(y = "On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0(unique(fancf_screen_ratio$guide), " ",
                 unique(fancf_screen_ratio$location_OT)))

## make Fig. S1b data frame for source data file
fig_s1b1_source <- fancf_screen %>%
  group_by(Experiment) %>%
  mutate(Panel = "b") %>%
  ungroup() %>%
  indel_source()

fig_s1b2_source <- fancf_screen_ratio %>%
  mutate(Panel = "b") %>%
  ratio_source()

fig_s1b_source <- left_join(fig_s1b1_source, fig_s1b2_source)

## append to source data list
source_data_list <- append(source_data_list,
                           list("FigS1b" = fig_s1b_source))
 
# ===========================================================================================

## make normalized ratio table and plot
norm_ratio <- dRNA_screens %>%
  filter(Experiment != 24 &
         Experiment != 13 &
         Experiment != 14 &
         Experiment != 11) %>%
  ratio_calc() %>%
  group_by(Experiment, X_lab) %>%
  mutate(avg = mean(ratio, na.rm = TRUE),
         n_ON = sum(!is.na(Indel_stringent_ON)),
         n_OT = sum(!is.na(Indel_stringent_OT)),
         avg_ON = mean(Indel_stringent_ON, na.rm = TRUE),
         avg_OT = mean(Indel_stringent_OT, na.rm = TRUE),
         sem_ON = sd(Indel_stringent_ON, na.rm = TRUE) / sqrt(n_ON),
         sem_OT = sd(Indel_stringent_OT, na.rm = TRUE) / sqrt(n_OT),
         sem_ratio = (avg_ON / avg_OT) * sqrt((sem_ON / avg_ON)^2 + (sem_OT / avg_OT)^2)) %>%
  ungroup() %>%
  group_by(Experiment) %>%
  mutate(norm_ratio = avg / first(avg),
         sem_norm_ratio = (avg / first(avg) *
                             sqrt((sem_ratio / avg)^2 +
                                    (first(sem_ratio) / first(avg))^2)),
         ON_ratio = avg_ON / first(avg_ON),
         sem_ON_ratio = (avg_ON / first(avg_ON) *
                           sqrt((sem_ON / avg_ON)^2 +
                                  (first(sem_ON) / first(avg_ON))^2)),
         OT_ratio = avg_OT / first(avg_OT),
         test_norm_ratio = ratio / first(avg),
         test_on_ratio = Indel_stringent_ON / first(avg_ON),
         test_ot_ratio = Indel_stringent_OT / first(avg_OT))

## count median dRNAs tested per site
median_dRNA <- norm_ratio %>%
  filter(X_lab != "- - -" &
         X_lab != "+ + -" &
         !is.na(location_OT)) %>%
  ungroup() %>%
  group_by(guide, location_OT) %>%
  summarise(count = n_distinct(X_lab)) %>%
  ungroup() %>%
  summarise(median(count))

## create table, filter
norm_table <- norm_ratio %>%
  ## filter dRNAs that reduce cutting at ON target by half
  filter(ON_ratio > 0.5) %>% 
  ## select max normalized ratio
  filter(norm_ratio == max(norm_ratio)) %>% 
  ## filter out ratio < 2
  filter(norm_ratio > 2) %>%
  mutate(best_dRNA = X_lab) %>%
  unite(Site, guide, location_OT, sep = " ") %>%
  ## grouping for facet plot in 1c
  mutate(group = 1,
         best_dRNA = gsub("\\+ \\+ ", "", best_dRNA)) %>%
  ## arrange in descending order
  arrange(desc(norm_ratio))

## make second table, using only missing experiments
norm_table_2 <- norm_ratio %>%
  unite(Site, guide, location_OT, sep = " ") %>%
  ## remove only missing experiemnt
  filter(!Site %in% norm_table$Site) %>%
  mutate(best_dRNA = X_lab) %>%
  ## filter max norm ratio for missing sites
  filter(norm_ratio == max(norm_ratio)) %>%
  ## groups on whether ON editing is < 0.5 or not
  mutate(group = ifelse(ON_ratio < 0.5, 2, 3),
         best_dRNA = gsub("\\+ \\+ ", "", best_dRNA)) %>%
  group_by(Site) %>%
  ## if more than one experiment for same site, select max
  filter(norm_ratio == max(norm_ratio)) %>%
  arrange(desc(norm_ratio)) # arrange top to bottom

## create source file for fig 1c 
fig_1c_source <- full_join(norm_table, norm_table_2) %>%
  mutate(Panel = "c") %>%
  ungroup() %>%
  separate(col = Site, into = c("guide", "location"), sep = " (?=[^ ]+$)") %>%
  select(Panel, Name, guide, location, ratio, test_norm_ratio, Indel_stringent_ON,
         test_on_ratio, Indel_stringent_OT, test_ot_ratio) %>%
  rename("Guide" = guide,
         "Off-target" = location,
         "ON:OT ratio" = ratio,
         "Normalized ON:OT ratio" = test_norm_ratio,
         "Indels ON" = Indel_stringent_ON,
         "Normalized ON ratio" = test_on_ratio,
         "Indels OT" = Indel_stringent_OT,
         "Normalized OT ratio" = test_ot_ratio)

## append to source data list
source_data_list <- append(source_data_list, list("Fig1c" = fig_1c_source))

## join tables together 
norm_table_2 <- full_join(norm_table, norm_table_2) %>%
  ungroup() %>%
  select(Experiment, Site, X_lab, n_ON, norm_ratio,
         ON_ratio, OT_ratio, sem_norm_ratio, sem_ON_ratio, group) %>%
  select(-c(Experiment)) %>%
  rename(n = n_ON) %>%
  unique()

## create Figure 1c - normalized ratio plot with sem
fig_1c <- ggplot(data = norm_table_2, aes(x = fct_inorder(Site), y = norm_ratio, fill = factor(group))) +
  geom_errorbar(stat = "identity",
                aes(ymin = norm_ratio - sem_norm_ratio, 
                    ymax = norm_ratio + sem_norm_ratio),
                alpha = 0.7, width = 0.3) +
  geom_point(stat = "identity", size = 2.5, pch = 21, alpha = 0.7) +
  facet_grid(~ group, scales = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text()) +
  scale_fill_manual(values = pal[3:5],
                     labels = c("successful", "reduced on-target editing",
                                "insufficient off-target suppression")) +
  scale_y_continuous(trans = "log2",
                     breaks = c(1, 2, 4, 8, 16, 32, 64),
                     labels = c(1, 2, 4, 8, 16, 32, 64),
                     expand = c(0, 0),
                     limits = c(min(norm_table_2$norm_ratio - norm_table_2$sem_norm_ratio) * 0.9, 64)) +
  xlab("Off-target site") +
  ylab("Specificity ratio improvement\n(fold change)")

## Remove column from table for printing
norm_table_3 <- norm_table_2 %>%
  select(-group, -OT_ratio)

## Write csv file, Table S1
write_csv(norm_table_3, "Stat_analysis/Supp_tables/TableS1_dRNA_screen_summary.csv")

## print table
kable(norm_table_2) %>%
  kable_styling(latex_options="scale_down")

## summarize Figure 1c for paper text
norm_results <- summarise(ungroup(norm_table),
                     n = n(),
                     mean_ratio = mean(norm_ratio),
                     mean_on = mean(ON_ratio),
                     sem_ratio = sd(norm_ratio) / sqrt(n),
                     sem_on = sd(ON_ratio) / sqrt(n))

```

```{r in vitro gels}

## import in vitro cleavage data
in_vitro2 <- read_csv("Data/invitro_gels_2.csv") %>%
  group_by(Exp, Site) %>%
  mutate(uncut = Peak_1 / (Peak_1 + Peak_2 + Peak_3),
         addition_order = if_else(is.na(addition_order), "-", addition_order),
         addition_order = if_else(addition_order == "GD", "sgRNA", addition_order),
         addition_order = if_else(addition_order == "DG", "dRNA", addition_order)) %>%
  unite("X_lab", c("dRNA", "guide", "addition_order"), sep = " ") %>%
  select(-substrate)

## split into on and off target
iv2_on <- filter(in_vitro2, Site == "FANCFg2 ON" &
                 word(X_lab, 3) == "-") %>%
  mutate(X_lab = gsub(" -", "", X_lab))

iv2_ot <- filter(in_vitro2, Site == "FANCFg2 OT1" &
                 word(X_lab, 3) == "-") %>%
  mutate(X_lab = gsub(" -", "", X_lab))

## filter pre-incubation experiments
iv2_on_pre <- filter(in_vitro2, Site == "FANCFg2 ON" &
                     word(X_lab, 3) != "-")

iv2_ot_pre <- filter(in_vitro2, Site == "FANCFg2 OT1" &
                     word(X_lab, 3) != "-")

## plot figure 2b
fig_2b <- ggplot(iv2_on, aes(fct_inorder(X_lab), uncut)) +
  geom_beeswarm(cex = 2, fill = pal[1], size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = iv2_on, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.5)) + # plots mean of each site
  scale_x_discrete(breaks = unique(iv2_on$X_lab),
                   labels = addline_format(unique(iv2_on$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 1),
                     breaks = c(0, 0.2, 0.4, 0.6, 0.8, 1),
                     labels = c(0, 0.2, 0.4, 0.6, 0.8, 1)) +
  ylab("Fraction uncut") +
  ggtitle("FANCF sgRNA2 on-target\n")

## plot figure 2a
fig_2a <- ggplot(iv2_ot, aes(fct_inorder(X_lab), uncut)) +
  geom_beeswarm(cex = 1, fill = pal[2], size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = iv2_ot, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.5)) + # plots mean of each site
  scale_x_discrete(breaks = unique(iv2_ot$X_lab),
                   labels = addline_format(unique(iv2_ot$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 1),
                     breaks = c(0, 0.2, 0.4, 0.6, 0.8, 1.0),
                     labels = c(0, 0.2, 0.4, 0.6, 0.8, 1.0)) +
  ylab("Fraction uncut") +
  ggtitle("FANCF sgRNA2 off-target\n")

## create labels for Figure 2
fig_2_label <- label_plot(data = iv2_on,
                            aes(x = Site,
                                y = uncut)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\ndRNA-RNP (fmoles)\n\nsgRNA-RNP (fmoles)")

## bring in annotated gel image for panel a 
fig_2a_gel <- ggdraw() +
  ggtitle("a") +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  draw_image(magick::image_read_pdf("Images/fig2a_off.pdf", density = 600))

## bring in annotated gel image for panel b
fig_2b_gel <- ggdraw() + 
  labs(title = "b") + 
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  draw_image(magick::image_read_pdf("Images/fig2b_on.pdf", density = 600))

## arrange figure 2
fig_2 <- fig_2a_gel + fig_2_label + plot_spacer() + fig_2a +
  fig_2b_gel + fig_2_label + plot_spacer() + fig_2b +
  plot_layout(ncol = 4, widths = c(2, 0.05, 0.25, 1.25)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 30, l = 7))

## save figure 2 - in vitro cutting
ggsave("Stat_analysis/Main_figures/Fig2_mechanism.pdf", fig_2, 
       height = 12, width = 14)

# ===========================================================================================

## plot pre-incubation in vitro data - on target
fig_s8b <- ggplot(iv2_on_pre, aes(fct_inorder(X_lab), uncut)) +
  geom_beeswarm(cex = 1.2, fill = pal[1], size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = iv2_on_pre, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.5)) + # plots mean of each site
  scale_x_discrete(breaks = unique(iv2_on_pre$X_lab),
                   labels = addline_format(unique(iv2_on_pre$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 1),
                     breaks = c(0, 0.2, 0.4, 0.6, 0.8, 1),
                     labels = c(0, 0.2, 0.4, 0.6, 0.8, 1)) +
  ylab("Fraction uncut") +
  ggtitle("FANCF sgRNA2 on-target")

## plot pre-incubation in vitro data - off target
fig_s8a <- ggplot(iv2_ot_pre, aes(fct_inorder(X_lab), uncut)) +
  geom_beeswarm(cex = 1.2, fill = pal[2], size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = iv2_ot_pre, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.5)) + # plots mean of each site
  scale_x_discrete(breaks = unique(iv2_ot_pre$X_lab),
                   labels = addline_format(unique(iv2_ot_pre$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 1),
                     breaks = c(0, 0.2, 0.4, 0.6, 0.8, 1),
                     labels = c(0, 0.2, 0.4, 0.6, 0.8, 1)) +
  ylab("Fraction uncut") +
  ggtitle("FANCF sgRNA2 off-target")

## bring in annotated gel image for panel a 
fig_s8a_gel <- ggdraw() +
  ggtitle("a") +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  draw_image(magick::image_read_pdf("Images/figs8a_off.pdf", density = 600))

## bring in annotated gel image for panel b 
fig_s8b_gel <- ggdraw() +
  ggtitle("b") +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  draw_image(magick::image_read_pdf("Images/figs8b_on.pdf", density = 600))

## make fig s8 labels
fig_s8_label <- label_plot(data = iv2_on_pre,
                           aes(x = Site,
                               y = uncut)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\ndRNA-RNP (fmoles)\n\nsgRNA-RNP (fmoles)\n\npreincubation")

## plot layout for figure s8 - preincubation in vitro data
fig_s8 <- fig_s8a_gel + fig_s8_label + plot_spacer() + fig_s8a + 
  fig_s8b_gel + fig_s8_label + plot_spacer() + fig_s8b + 
  plot_layout(ncol = 4, widths = c(1, 0.05, 0.15, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 30, l = 7))

## save figure s8 - preincubation of dRNA or sgRNA
ggsave("Stat_analysis/Supp_figures/FigS8_preincubation.pdf", fig_s8,
       height = 12, width = 8)

## create source file for fig 2b
fig_2b_source <- iv2_on %>%
  ungroup() %>%
  mutate(Panel = "b",
         Site = gsub("g", " sgRNA ", Site)) %>%
  rename("Replicate" = Exp,
         "Fraction uncut" = uncut,
         "Uncut DNA" = Peak_1,
         "Cut DNA 1" = Peak_2,
         "Cut DNA 2" = Peak_3) %>%
  select(Panel, Lane, everything()) %>%
  select(-X_lab)

## create source file for fig 2a
fig_2a_source <- iv2_ot %>%
  ungroup() %>%
  mutate(Panel = "a", 
         Site = gsub("g", " sgRNA ", Site)) %>%
  rename("Replicate" = Exp,
         "Fraction uncut" = uncut,
         "Uncut DNA" = Peak_1,
         "Cut DNA 1" = Peak_2,
         "Cut DNA 2" = Peak_3) %>%
  select(Panel, Lane, everything()) %>%
  select(-X_lab)

## create source file for fig S8b
fig_s8b_source <- iv2_on_pre %>%
  ungroup() %>%
  mutate(Panel = "b", 
         Site = gsub("g", " sgRNA ", Site)) %>%
  rename("Replicate" = Exp,
         "Fraction uncut" = uncut,
         "Uncut DNA" = Peak_1,
         "Cut DNA 1" = Peak_2,
         "Cut DNA 2" = Peak_3,
         "X label" = X_lab) %>%
  select(Panel, Lane, Replicate, Site, "X label", everything())

## create source file for fig S8a
fig_s8a_source <- iv2_ot_pre %>%
  ungroup() %>%
  mutate(Panel = "a", 
         Site = gsub("g", " sgRNA ", Site)) %>%
  rename("Replicate" = Exp,
         "Fraction uncut" = uncut,
         "Uncut DNA" = Peak_1,
         "Cut DNA 1" = Peak_2,
         "Cut DNA 2" = Peak_3,
         "X label" = X_lab) %>%
  select(Panel, Lane, Replicate, Site, "X label", everything())

## append to source data list
source_data_list <- append(source_data_list,
                           list("Fig2a - plot" = fig_2a_source,
                                "Fig2b - plot" = fig_2b_source,
                                "FigS8a - plot" = fig_s8a_source,
                                "FigS8b - plot" = fig_s8b_source))


```

```{r ciCas9}
## read in csv file
cicas9 <- read_csv("Data/cicas9.csv") %>%
  filter(Time != 24 &
         Treated != "DMSO" &
         Reads_passing >= 1000) %>%
  mutate(Name = gsub("OT1d1", "dRNA1", Name),
         Name = gsub("FANCFg2", "FANCF sgRNA2", Name),
         Name = gsub("dNT", "non-targeting dRNA", Name),
         Name = gsub("cells", "NT", Name),
         Site = gsub("FANCFg2 ON", "on-target", Site),
         Site = gsub("FANCFg2 OT1", "off-target", Site),
         Name = factor(Name, levels = c("FANCF sgRNA2", "FANCF sgRNA2 + non-targeting dRNA",
                                        "FANCF sgRNA2 + dRNA1", "dRNA1", "NT")),
         Site = factor(Site, levels = c("on-target", "off-target")))

## subset to on and off target
cicas9_fig_3_on <- filter(cicas9, Treated == "A115" &
                         Reads_passing >= 1000 &
                         Site == "on-target")

cicas9_fig_3_ot <- filter(cicas9, Treated == "A115" &
                         Reads_passing >= 1000 &
                         Site == "off-target")

## plot limited cicas9
fig_3a1 <- ggplot(data = cicas9_fig_3_on,
                  aes(x = Time, y = Indel_stringent,
                      linetype = Name, fill = Name)) +
  stat_summary(data = cicas9_fig_3_on, fun.data = "mean_se",
               geom = "errorbar", width = 0.2, linetype = "solid") +
  stat_summary(data = cicas9_fig_3_on, fun = "mean",
               geom = "line") +
  stat_summary(data = cicas9_fig_3_on, fun = "mean", alpha = 0.7,
               geom = "point", size = 2.5, pch = 21, color = "black") +
  scale_y_continuous(limits = c(-0.5, 10),
                     expand = c(0, 0),
                     breaks = c(0, 2, 4, 6, 8, 10),
                     labels = c(0, 2, 4, 6, 8, 10)) +
  scale_x_continuous(limits = c(-0.5, 16.5),
                     expand = c(0, 0),
                     breaks = c(0, 2, 4, 6, 8, 10, 12, 14, 16),
                     labels = c(0, 2, 4, 6, 8, 10, 12, 14, 16)) +
  scale_fill_manual(values = pal) +
  scale_color_manual(values = pal) +
  theme(axis.title.x = element_text(),
        legend.position = c(0.05, 1.01),
        legend.justification = c(0, 1),
        legend.background = element_rect(color = NA, fill = NA),
        legend.key = element_rect(color = NA, fill = NA),
        legend.box = "horizontal",
        legend.spacing.x = unit(1, "pt")) +
  labs(x = "Time (h)",
       y = "Indel frequency",
       title = "FANCF sgRNA2 on-target") +
  guides(linetype = guide_legend(unit(2, "lines"), order = 1, label = FALSE),
         fill = guide_legend(order = 0))
  
## plot limited cicas9
fig_3a2 <- ggplot(data = cicas9_fig_3_ot,
                  aes(x = Time, y = Indel_stringent,
                      linetype = Name, fill = Name)) +
  stat_summary(data = cicas9_fig_3_ot, fun.data = "mean_se",
               geom = "errorbar", width = 0.2, linetype = "solid") +
  stat_summary(data = cicas9_fig_3_ot, fun = "mean",
               geom = "line") +
  stat_summary(data = cicas9_fig_3_ot, fun = "mean", alpha = 0.7,
               geom = "point", size = 2.5, pch = 21, color = "black") +
  scale_y_continuous(limits = c(-0.5, 6),
                     expand = c(0, 0),
                     breaks = c(0, 2, 4, 6),
                     labels = c(0, 2, 4, 6)) +
  scale_x_continuous(limits = c(-0.5, 16.5),
                     expand = c(0, 0),
                     breaks = c(0, 2, 4, 6, 8, 10, 12, 14, 16),
                     labels = c(0, 2, 4, 6, 8, 10, 12, 14, 16)) +
  scale_fill_manual(values = pal) +
  scale_color_manual(values = pal) +
  theme(axis.title.x = element_text(),
        legend.position = c(0.05, 1.01),
        legend.justification = c(0, 1),
        legend.background = element_rect(color = NA, fill = NA),
        legend.key = element_rect(color = NA, fill = NA),
        legend.box = "horizontal",
        legend.spacing.x = unit(1, "pt")) +
  labs(x = "Time (h)",
       y = "Indel frequency",
       title = "FANCF sgRNA2 off-target") +
  guides(linetype = guide_legend(unit(2, "lines"), order = 1, label = FALSE),
         fill = guide_legend(order = 0))

## create source file for fig 3a
fig_3a_source <- cicas9 %>%
  mutate(Panel = "a",
         Guide = "FANCF sgRNA 2",
         ot = "OT1") %>%
  select(Site, Name, Replicate, Time, Indel_stringent, Panel, Guide, ot) %>%
  pivot_wider(names_from = Site,
              values_from = Indel_stringent) %>%
  select(Panel, Name, Replicate, Guide, ot, "on-target", "off-target") %>%
  rename("Indels ON" = "on-target",
         "Indels OT" = "off-target",
         "Off-target" = ot) 

## append to source data list
source_data_list <- append(source_data_list,
                           list("Fig3a" = fig_3a_source))

```

```{r titrations}
titrations_all <- read_csv("Data/titrations.csv") %>%
  filter(Reads_passing >= 1000 &
           Experiment != 2) %>%
  mutate(guide = word(Site, 1),
         target = word(Site, -1),
         Site = gsub("g", " sgRNA", Site),
         guide = gsub("g", " sgRNA", guide),
         Experiment = as_factor(Experiment))

titrations <- titrations_all %>%
  group_by(Experiment) %>%
  mutate(tag = letters[Experiment]) %>%
  nest() %>%
  mutate(plot1 = map2(data, Experiment, ~ggplot(data = .x) +
                       aes(x = X_lab, y = Indel_stringent, linetype = Site, fill = Site) +
                       stat_summary(data = .x, fun.data = "mean_se",
                                    geom = "errorbar", width = 0.05, linetype = "solid") +
                       stat_summary(data = .x, fun = "mean", geom = "line") +
                       stat_summary(data = .x, fun = "mean", alpha = 0.7,
                                    geom = "point", size = 2.5, pch = 21, color = "black") +
                       scale_fill_manual(values = pal,
                                         breaks = unique(.$Site),
                                         labels = c("on-target", "off-target")) +
                       coord_cartesian(ylim =
                                         if(max(.$Indel_stringent) > 25) {c(0, 30)}
                                         else {c(0, 20)},
                                       clip = "off") +
                       scale_y_continuous(expand = c(0, 0),
                                          breaks =
                                            if(max(.$Indel_stringent) > 25) {seq(0, 30, by = 5)}
                                            else {seq(0, 20, by = 5)}) +
                       scale_x_continuous(expand = c(0, 0), limits = c(-0.1, 4.2)) +
                       theme(axis.title.x = element_text(),
                             legend.position = c(0.7, 1),
                             legend.justification = c(0, 1),
                             legend.background = element_rect(color = NA, fill = NA),
                             legend.key = element_rect(color = NA, fill = NA),
                             legend.box = "horizontal",
                             legend.direction = "vertical",
                             legend.spacing.x = unit(1, "pt")) +
                       labs(x = "[dRNA] / [sgRNA]",
                            y = "Indel frequency (%)",
                            title = paste0(unique(.$Site)[2]), "\n") +
                       guides(linetype = guide_legend(unit(2, "lines"),
                                                      order = 1, label = FALSE),
                              fill = guide_legend(order = 0))),
         label_plots = map2(data, Experiment,
                           ~label_plot(data = .x) +
                             aes(Site,
                                 Indel_stringent) +
                             geom_point(color = "NA") +
                             labs(title = .$tag) +
                             theme(axis.text.x = element_blank())))

## calculate ratio for titrations over all concentrations
titrations_ratio <- titrations_all %>%
  ratio_calc() %>%
  group_by(Experiment) %>%
  nest() %>%
  mutate(plot2 = map2(data, Experiment, ~ggplot(data = .x) +
                        aes(X_lab, ratio) +
                        stat_summary(data = .x, fun.data = "mean_se",
                                    geom = "errorbar", width = 0.05, linetype = "solid") +
                        stat_summary(data = .x, fun = "mean", geom = "line") +
                        stat_summary(data = .x, fun = "mean", alpha = 0.7,
                                    geom = "point", size = 2.5, pch = 21,
                                    color = "black", fill = pal[5]) +
                        coord_cartesian(clip = "off",
                                        ylim = if (max(.$ratio) < 70) {c(0, 60)}
                                        else if (max(.$ratio) > 600) {c(0, 600)}
                                        else {c(0, 250)}) +
                        scale_y_continuous(expand = c(0, 0),
                                           breaks = if (max(.$ratio) < 70) {seq(0, 60, by = 10)}
                                           else if (max(.$ratio) > 600) {seq(0, 600, by = 100)}
                                           else {seq(0, 250, by = 50)}) +
                        scale_x_continuous(expand = c(0, 0), limits = c(-0.1, 4.15)) +
                        theme(axis.title.x = element_text()) +
                        labs(x = "[dRNA] / [sgRNA]",
                             y = "On-target/Off-target\nindel frequency ratio",
                             title = paste0(.$guide, " ", .$location_OT), "\n")))

## join indels and ratio plots
joined_titrations <- inner_join(titrations, titrations_ratio, by = "Experiment") %>%
  ## join to plot side by side
   pivot_longer(cols = c("label_plots", "plot1", "plot2"),
                names_to = "plottype",
                values_to = "plot")

## arrange into plot s9 - additonal titrations
fig_s9 <- wrap_plots(joined_titrations$plot, ncol = 3, widths = c(0.05, 1, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

## save fig s9 - additional titrations
ggsave("Stat_analysis/Supp_figures/FigS9_titrations.pdf", fig_s9,
       height = 16, width = 12)

## create source file for fig s9
fig_s9_indels_source <- titrations_all %>%
  group_by(Experiment) %>%
  mutate(Panel = letters[Experiment]) %>%
  ungroup() %>%
  indel_source()

fig_s9_ratio_source <- titrations_all %>%
  ratio_calc() %>%
  group_by(Experiment) %>%
  mutate(Panel = letters[Experiment]) %>%
  ungroup() %>%
  ratio_source()

fig_s9_source <- left_join(fig_s9_indels_source, fig_s9_ratio_source)

## split list by panel 
fig_s9_list <- fig_s9_source %>%
  named_group_split(Panel)

## change names to figure panels
names(fig_s9_list) <- paste0("FigS9", names(fig_s9_list))

## append to source data list
source_data_list <- append(source_data_list, fig_s9_list)

# ===========================================================================================

## Import titration data for VEGFAg3 OT2
v3_titration <- read_csv("Data/titrations.csv") %>%
  filter(Reads_passing >= 1000 &
         Experiment == 2) %>%
  mutate(target = word(Site, -1),
         Site = gsub("g", " sgRNA", Site),
         guide = word(Site, start = 1, end = 2),
         Panel = "b")


## calculate ratio
v3_titration_ratio <- ratio_calc(v3_titration) %>%
  mutate(Site = paste0(guide, location_ON),
         Panel = "b")

## Plot data
fig_3b1 <- ggplot(data = v3_titration, aes(x = X_lab, y = Indel_stringent,
                                           fill = Site)) +
  stat_summary(data = v3_titration, fun.data = "mean_se",
               geom = "errorbar", width = 0.05, linetype = "solid") +
  stat_summary(data = v3_titration, fun = "mean", geom = "line",
               aes(linetype = Site)) +
  stat_summary(data = v3_titration, fun = "mean", alpha = 0.7,
               geom = "point", size = 2.5, pch = 21, color = "black") +
  scale_fill_manual(values = pal,
                    breaks = unique(v3_titration$Site),
                    labels = c("on-target", "off-target")) +
  theme(axis.title.x = element_text(),
        legend.position = c(0.05, 1),
        legend.justification = c(0, 1),
        legend.background = element_rect(color = NA, fill = NA),
        legend.key = element_rect(color = NA, fill = NA),
        legend.box = "horizontal",
        legend.direction = "vertical",
        legend.spacing.x = unit(1, "pt")) +
  scale_y_continuous(expand = c(0, 0), limits = c(-0.5, round_max(v3_titration)),
                     breaks = seq(0, round_max(v3_titration), by = round_max(v3_titration) / 5)) +
  scale_x_continuous(expand = c(0, 0), limits = c(-0.1, 4.15)) +
  labs(x = "[dRNA] / [sgRNA]",
       y = "Indel frequency (%)",
       title = paste0(unique(v3_titration$Site)[2]), "\n") +
  guides(linetype = guide_legend(unit(2, "lines"), order = 1, label = FALSE),
         fill = guide_legend(order = 0))

fig_3b2 <- ggplot(data = v3_titration_ratio, aes(X_lab, ratio)) +
  stat_summary(data = v3_titration_ratio, fun.data = "mean_se",
               geom = "errorbar", width = 0.05, linetype = "solid") +
  stat_summary(data = v3_titration_ratio, fun = "mean", geom = "line") +
  stat_summary(data = v3_titration_ratio, fill = pal[5], fun = "mean", alpha = 0.7,
               geom = "point", size = 2.5, pch = 21, color = "black") +
  theme(axis.title.x = element_text()) +
  scale_y_continuous(expand = c(0, 0), limits = c(-0.5, round_max(v3_titration_ratio)),
                     breaks = seq(0, round_max(v3_titration_ratio),
                                  by = round_max(v3_titration_ratio) / 5)) +
  scale_x_continuous(expand = c(0, 0), limits = c(-0.1, 4.15)) +
  labs(x = "[dRNA] / [sgRNA]",
       y = "On-target/Off-target\nindel frequency ratio",
       title = paste0(unique(v3_titration$Site)[2]), "\n")

## create labels
fig_3_label <- label_plot(data = v3_titration_ratio,
                            aes(x = guide,
                                y = ratio)) +
  geom_point(color = "NA") +
  theme(axis.text.x = element_blank())

fig_3a_label <- fig_3_label + 
  labs(title = "a")

fig_3b_label <- fig_3_label +
  labs(title = "b")

## arrange fig 3 - cicas9 and titrations
fig_3 <- fig_3a_label + fig_3a1 + fig_3a2 +
  fig_3b_label + fig_3b1 + fig_3b2 +
  plot_layout(ncol = 3, widths = c(0.05, 1, 1))

## save fig 3
ggsave("Stat_analysis/Main_figures/Fig3_cicas9_and_titrations.pdf", fig_3, 
       height = 12, width = 12)

## create source file for fig 3b
fig_3b_indels_source <- v3_titration %>%
  indel_source()

fig_3b_ratio_source <- v3_titration_ratio %>%
  ratio_source()

fig_3b_source <- left_join(fig_3b_indels_source, fig_3b_ratio_source)

## append to source data list
source_data_list <- append(source_data_list,
                           list("Fig3b" = fig_3b_source))

```

```{r truncated guides}

## import truncated VEGFA guide 
tru_vegfa <- dRNA_screens %>%
  filter(Experiment == 24 &
         Reads_passing >= 1000) %>%
  mutate(facet_group = ifelse(Indel_stringent >= 5, 1, 2),
         ymin = ifelse(facet_group == 1, 14, 0),
         ymax = ifelse(facet_group == 1, 20, 3),
         X_lab = str_replace(X_lab, "^[+]", "WT"),
         Site = gsub("truVEGFA ", "VEGFA tru-", Site),
         X_lab = gsub("[0-9]", "+", X_lab),
         guide = word(Site, start = 1, end = 2),
         target = word(Site, -1),
         Panel = "a")

## ratio calculations
tru_vegfa_ratio <- ratio_calc(tru_vegfa)

## plot tru-VEGFA indels
fig_4a1 <- ggplot(data = tru_vegfa, aes(X_lab, Indel_stringent, fill = Site)) +
  aes(x = fct_inorder(X_lab)) +
  geom_blank(aes(y = ymin)) +
  geom_blank(aes(y = ymax)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21, alpha = 0.7) +
  stat_summary(data = tru_vegfa, fun = "mean",
               geom = "point", pch = "_", size = 8) + # plots mean of each site
  facet_grid(facet_group ~ ., scales = "free") +
  theme(strip.background = element_blank(),
        strip.text = element_blank(),
        panel.spacing = unit(1.2, "lines")) +
  scale_x_discrete(labels = addline_format(unique(tru_vegfa$X_lab))) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = pal, breaks = unique(tru_vegfa$Site),
                     labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(unique(tru_vegfa$Site)[2])

## plot tru-VEGFA ratio
fig_4a2 <- ggplot(data = tru_vegfa_ratio, aes(X_lab, ratio)) +
  aes(x = fct_inorder(X_lab)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21,
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = tru_vegfa_ratio, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.5)) + 
  scale_x_discrete(labels = addline_format(unique(tru_vegfa_ratio$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 60),
                     breaks = seq(0, 60, by = 20)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0(unique(tru_vegfa_ratio$guide), " ",
                 unique(tru_vegfa_ratio$location_OT)))

fig_4a_labels <- label_plot(tru_vegfa, aes(x = as_factor(Experiment),
                                           y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = "\nCas9\n\ntru-VEGFA sgRNA3\n\ndRNA")

fig_4a1_label <- fig_4a_labels +
  labs(title = "a")

## create source file for fig 4a
fig_4a_indels_source <- tru_vegfa %>%
  indel_source()

fig_4a_ratio_source <- tru_vegfa_ratio %>%
  mutate(Panel = "a") %>%
  ratio_source()

fig_4a_source <- left_join(fig_4a_indels_source, fig_4a_ratio_source)

## append to source data list
source_data_list <- append(source_data_list,
                           list("Fig4a" = fig_4a_source))

```

```{r hifi}
## read hifi .csv file into memory
hifi <- read_csv("Data/hifi_dots.csv") %>%
  filter(Reads_passing >= 1000 &
         !grepl("4", X_lab)) %>% # removes 4x dRNA concentrations
  mutate(Cas9 = factor(Cas9, levels = c("WT", "ECas9", "HF1", "Hypa")),
         Site = gsub("g", " sgRNA", Site),
         X_lab = gsub("[0-9]", "+", X_lab),
         X_lab = gsub("HF\\+", "HF1", X_lab),
         guide = word(Site, start = 1, end = 2),
         target = word(Site, -1))

## filter data and create labels for plots
V3_hifi <- hifi %>%
  filter(Experiment == 2) %>%
  mutate(X_lab = gsub("\\s1", " 2", X_lab))

F2_hifi <- hifi %>%
  filter(Experiment == 1) %>%
  mutate(Panel = "b")

labels_V3_hifi <- V3_hifi %>%
  distinct(X_lab)

labels_F2_hifi <- F2_hifi %>%
  distinct(X_lab)

# ===========================================================================================

## generate ratio for each sample
hifi_ratio <- hifi %>%
  ratio_calc_hifi()

V3_hifi_ratio <- hifi_ratio %>%
  filter(Experiment == 2)

F2_hifi_ratio <- hifi_ratio %>%
  filter(Experiment == 1) %>%
  mutate(Panel = "b")

ratio_labels_V3_hifi <- labels_V3_hifi %>%
  filter(X_lab != "- - -")

ratio_labels_F2_hifi <- labels_F2_hifi %>%
  filter(X_lab != "- - -")

# ===========================================================================================

## Figure 4b - FANCFg2 OT1 with enhanced specificity Cas9s
fig_4b1 <- ggplot(data = F2_hifi, aes(factor(X_order), Indel_stringent, fill = Site)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21, alpha = 0.7) +
  stat_summary(data = F2_hifi, fun = "mean",
               geom = "point", pch = "_", size = 8) + # plots mean of each site
  facet_grid(~ Cas9, scales = "free", space = "free") +
  annotate("segment", x = -Inf, xend = Inf, y = -Inf, yend = -Inf) +
  annotate("segment", x = -Inf, xend = -Inf, y = -Inf, yend = Inf) +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank()) +
  scale_x_discrete(breaks = unique(F2_hifi$X_order),
                   labels = addline_format(labels_F2_hifi$X_lab)) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 25),
                     breaks = seq(0, 25, by = 5)) +
  scale_fill_manual(values = pal,
                     breaks = unique(F2_hifi$Site),
                     labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(unique(F2_hifi$Site)[2])

fig_4b2 <- ggplot(data = F2_hifi_ratio, aes(factor(X_order), ratio)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21,
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = F2_hifi_ratio, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  facet_grid(~ Cas9, scales = "free", space = "free") +
  annotate("segment", x = -Inf, xend = Inf, y = -Inf, yend = -Inf) +
  annotate("segment", x = -Inf, xend = -Inf, y = -Inf, yend = Inf) +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank()) +
  scale_x_discrete(breaks = unique(F2_hifi_ratio$X_order[!is.na(F2_hifi_ratio$X_order)]),
                   labels = addline_format(unique(ratio_labels_F2_hifi$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 250),
                     breaks = seq(0, 250, by = 50)) +
  annotate("segment", x = -Inf, xend = Inf, y = -Inf, yend = -Inf) +
  annotate("segment", x = -Inf, xend = -Inf, y = -Inf, yend = Inf) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0(unique(F2_hifi_ratio$guide), " ", unique(F2_hifi_ratio$location_OT)))

fig_4b_labels <- label_plot(F2_hifi, aes(x = as_factor(Experiment),
                                           y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = label_grab(F2_hifi))

fig_4b1_label <- fig_4b_labels +
  labs(title = "b") 

## align
fig_4 <- (fig_4a1_label + plot_spacer() + fig_4a1 +
            fig_4a_labels + fig_4a2 +
            plot_layout(ncol = 5, widths = c(0.05, 0.1, 1, 0.05, 1))) /
  (fig_4b1_label + plot_spacer() + fig_4b1 +
     fig_4b_labels + fig_4b2 +
     plot_layout(ncol = 5, widths = c(0.05, 0.1, 1, 0.05, 1))) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))
  
  ## save fig 4 - hifi
ggsave("Stat_analysis/Main_figures/Fig4_hifi.pdf", fig_4,
       height = 12, width = 18)

## create source file for fig 4b
fig_4b_source <- F2_hifi %>%
  mutate("Off-target" = "OT1") %>%
  select(Panel, Name, guide, "Off-target", Cas9, X_lab, Indel_stringent, target) %>%
  pivot_wider(names_from = target,
              values_from = Indel_stringent) %>%
  mutate(Ratio = ifelse(X_lab == "- - -", NA, ON / OT1)) %>%
  rename("Guide" = guide,
         "Indels ON" = ON,
         "Indels OT" = OT1)

## append to source data list
source_data_list <- append(source_data_list,
                           list("Fig4b" = fig_4b_source))

# ===========================================================================================

## Figure S10 - additional site with hifi
fig_s10a1 <- ggplot(data = V3_hifi,
                   aes(factor(X_order), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 3, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = V3_hifi, fun = "mean",
               geom = "point", pch = "_", size = 8) + 
  facet_grid(~ Cas9, scales = "free", space = "free") +
  annotate("segment", x = -Inf, xend = Inf, y = -Inf, yend = -Inf) +
  annotate("segment", x = -Inf, xend = -Inf, y = -Inf, yend = Inf) +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank()) +
  scale_x_discrete(breaks = unique(V3_hifi$X_order),
                   labels = addline_format(labels_V3_hifi$X_lab)) +
  scale_fill_manual(values = pal,
                     breaks = unique(V3_hifi$Site),
                     labels = c("on-target", "off-target")) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 30),
                     breaks = seq(0, 30, by = 10)) +
  ylab("Indel frequency (%)") +
  ggtitle(unique(V3_hifi$Site)[2])

## additional hifi ratio
fig_s10a2 <- ggplot(data = V3_hifi_ratio, aes(factor(X_order), ratio)) +
  geom_beeswarm(size = 2.5, cex = 3, pch = 21,
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = V3_hifi_ratio, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  facet_grid(~ Cas9, scales = "free", space = "free") +
  annotate("segment", x = -Inf, xend = Inf, y = -Inf, yend = -Inf) +
  annotate("segment", x = -Inf, xend = -Inf, y = -Inf, yend = Inf) +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank()) +
  scale_x_discrete(breaks = unique(V3_hifi_ratio$X_order[!is.na(V3_hifi_ratio$X_order)]),
                   labels = addline_format(unique(ratio_labels_V3_hifi$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 50),
                     breaks = seq(0, 50, by = 10)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0(unique(V3_hifi_ratio$guide), " ", unique(V3_hifi_ratio$location_OT), "\n"))

## make labels
fig_s10_labels <- label_plot(V3_hifi, aes(x = as_factor(Experiment),
                                          y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(label = label_grab(V3_hifi))

## align
fig_s10 <- fig_s10_labels + fig_s10a1 + fig_s10_labels + fig_s10a2 +
  plot_layout(ncol = 2, widths = c(0.05, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7))

## save fig s9 - additional hifi
ggsave("Stat_analysis/Supp_figures/FigS10_hifi_V3.pdf", fig_s10,
       height = 10, width = 8)

## create source file for fig s10
fig_s10_source <- V3_hifi %>%
  mutate("Off-target" = "OT2") %>%
  select(Name, guide, "Off-target", Cas9, X_lab, Indel_stringent, target) %>%
  pivot_wider(names_from = target,
              values_from = Indel_stringent) %>%
  mutate(Ratio = ifelse(X_lab == "- - -", NA, ON / OT2)) %>%
  rename("Guide" = guide,
         "Indels ON" = ON,
         "Indels OT" = OT2)

## append to source data list
source_data_list <- append(source_data_list,
                           list("FigS10" = fig_s10_source))
```

```{r mdots}
## read mdots .csv file into memory
mdots <- read_csv("Data/mdots.csv") %>%
  mutate(target = word(Site, -1),
         Site = gsub("g", " sgRNA", Site),
         guide = word(Site, start = 1, end = 2),
         X_lab = gsub("[0-9]", "+", X_lab),
         X_lab = gsub("HF\\+", "HF1", X_lab),
         X_lab = gsub("\\+s", "+", X_lab))

## generate each mdots plot
V2_mdots <- filter(mdots, Reads_passing >= 1000 &
                   Experiment == 1) # filters out less than 1000 reads
V3_mdots <- filter(mdots, Reads_passing >= 1000 &
                   Experiment == 2)
Z1_mdots <- filter(mdots, Reads_passing >= 1000 &
                   Experiment == 3)
F2Z1_mdots <- filter(mdots, Reads_passing >= 1000 &
                     Experiment == 4)

V2_mdots_reduced <- filter(V2_mdots, X_lab == "+ + - - -" |
                        X_lab == "+ + + + +" |
                        X_lab == "- - - - -")

V2_mdots_reduced$X_lab <- str_replace(V2_mdots_reduced$X_lab, "^[+]", "WT")

eCas9_mdots <- filter(mdots, Reads_passing >= 1000 &
                        Experiment == 5)

hf1_mdots <- filter(mdots, Reads_passing >= 1000 &
                      Experiment == 6)

hypa_mdots <- filter(mdots, Reads_passing >= 1000 &
                       Experiment == 7)

# plot figure 5a - mdots indels
fig_5a1 <- ggplot(data = V2_mdots_reduced,
                  aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 2, dodge.width = 0.4, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = V2_mdots_reduced, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(width = 0.4)) +
  scale_x_discrete(labels = addline_format(unique(V2_mdots_reduced$X_lab))) +
  scale_fill_manual(values = pal,
                     breaks = unique(V2_mdots_reduced$Site),
                     labels = c("on-target", "OT1", "OT2", "OT17")) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 12),
                     breaks = seq(0, 12, by = 4)) +
  ylab("Indel frequency (%)") +
  ggtitle("VEGFA sgRNA2")

# plot fig 5b - hifi mdots indels
fig_5b1 <- ggplot(data = eCas9_mdots,
                  aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 2, dodge.width = 0.3, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = eCas9_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(width = 0.3)) +
  scale_x_discrete(labels = addline_format(unique(eCas9_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 15),
                     breaks = seq(0, 15, by = 5)) +
  scale_fill_manual(values = pal,
                     breaks = unique(eCas9_mdots$Site),
                     labels = c("on-target", "OT1", "OT2", "OT17")) +
  ylab("Indel frequency (%)") +
  ggtitle("VEGFA sgRNA2")

# plot figure s12a mdots for V2 - HF1
fig_s12a1 <- ggplot(data = hf1_mdots,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 2, dodge.width = 0.5, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = hf1_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(width = 0.5)) +
  scale_x_discrete(labels = addline_format(unique(hf1_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 3),
                     breaks = seq(0, 3, by = 1)) +
  scale_fill_manual(values = pal,
                     breaks = unique(hf1_mdots$Site),
                     labels = c("on-target", "OT1", "OT2", "OT17")) +
  ylab("Indel frequency (%)") +
  ggtitle("VEGFA sgRNA2")

# plot figure s12b mdots for V2 - hypa
fig_s12b1 <- ggplot(data = hypa_mdots,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 2, dodge.width = 0.5, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = hypa_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(width = 0.5)) +
  scale_x_discrete(labels = addline_format(unique(hypa_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 3),
                     breaks = seq(0, 3, by = 1)) +
  scale_fill_manual(values = pal,
                     breaks = unique(hypa_mdots$Site),
                     labels = c("on-target", "OT1", "OT2", "OT17")) +
  ylab("Indel frequency (%)") +
  ggtitle("VEGFA sgRNA2")

## plot indel frequency for VEGFAg2 mdots
fig_s11a1 <- ggplot(data = V2_mdots,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = -Inf, ymax = Inf,
           alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = -Inf, ymax = Inf,
           alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 5.5, xmax = 6.5,
           ymin = -Inf, ymax = Inf,
           alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 7.5, xmax = 8.5,
           ymin = -Inf, ymax = Inf,
           alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = V2_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.8)) +
  scale_x_discrete(labels = addline_format(unique(V2_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 12),
                     breaks = seq(0, 12, by = 3)) +
  scale_fill_manual(values = pal,
                     breaks = unique(V2_mdots$Site),
                     labels = c("on-target", "OT1", "OT2", "OT17")) +
  ylab("Indel frequency (%)") +
  ggtitle("VEGFA sgRNA2")

## plot indel frequency for VEGFAg3 mdots
fig_s11b1 <- ggplot(data = V3_mdots,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = -Inf, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = -Inf, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 5.5, xmax = 6.5,
           ymin = -Inf, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 7.5, xmax = 8.5,
           ymin = -Inf, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = V3_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.8)) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(V3_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 30),
                     breaks = seq(0, 30, by = 10)) +
  scale_fill_manual(values = pal,
                     breaks = unique(V3_mdots$Site),
                     labels = c("on-target", "OT2", "OT4", "OT18")) +
  ylab("Indel frequency (%)") +
  ggtitle("VEGFA sgRNA3")

## plot indel frequency for ZSCAN2g1 mdots
fig_s11c1 <- ggplot(data = Z1_mdots,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = -Inf, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = -Inf, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = Z1_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.8)) + 
  scale_x_discrete(labels = addline_format(unique(Z1_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 30),
                     breaks = seq(0, 30, by = 10)) +
  scale_fill_manual(values = pal,
                     breaks = unique(Z1_mdots$Site),
                     labels = c("on-target", "OT1", "OT2")) +
  ylab("Indel frequency (%)") +
  ggtitle("ZSCAN2 sgRNA1")

## plot indel frequency for FANCFg2 + ZSCAN2g1 mdots
fig_s11d1 <- ggplot(data = F2Z1_mdots,
                   aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 5.5, xmax = 6.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 7.5, xmax = 8.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 9.5, xmax = 10.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = F2Z1_mdots, fun = "mean",
               geom = "point", pch = "_", size = 8,
               position = position_dodge(0.8)) + 
  scale_x_discrete(labels = addline_format(unique(F2Z1_mdots$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 30),
                     breaks = seq(0, 30, by = 5)) +
  scale_fill_manual(values = c(pal[1], pal[3], pal[4], pal[2]),
                     breaks = unique(F2Z1_mdots$Site),
                     labels = c("FANCF sgRNA2 on-target",
                                "ZSCAN2 sgRNA1 on-target",
                                "FANCF sgRNA2 OT1",
                                "ZSCAN2 sgRNA1 OT2")) +
  ylab("Indel frequency (%)") +
  ggtitle("FANCF sgRNA2 and ZSCAN2 sgRNA1") +
  guides(fill = guide_legend(nrow = 2))

## generate ratio for each sample
V2_mdots_ratio <- ratio_calc(V2_mdots)

V2_mdots_ratio_reduced <- filter(V2_mdots_ratio, X_lab == "+ + - - -" |
                              X_lab == "+ + + + +" |
                              X_lab == "- - - - -") %>%
  mutate(X_lab = str_replace(X_lab, "^[+]", "WT"))

eCas9_mdots_ratio <- ratio_calc(eCas9_mdots) %>%
  filter(location_OT != "OT2") %>%
  group_by(Experiment, X_lab, location_OT) %>%
  mutate(avg = mean(ratio, na.rm = TRUE),
         n_ON = sum(!is.na(Indel_stringent_ON)),
         n_OT = sum(!is.na(Indel_stringent_OT)),
         avg_ON = mean(Indel_stringent_ON, na.rm = TRUE),
         avg_OT = mean(Indel_stringent_OT, na.rm = TRUE),
         sem_ON = sd(Indel_stringent_ON, na.rm = TRUE) / sqrt(n_ON),
         sem_OT = sd(Indel_stringent_OT, na.rm = TRUE) / sqrt(n_OT),
         sem_ratio = (avg_ON / avg_OT) * sqrt((sem_ON / avg_ON)^2 + (sem_OT / avg_OT)^2)) %>%
  ungroup()

hf1_mdots_ratio <- ratio_calc(hf1_mdots) %>%
  filter(location_OT != "OT2")

hypa_mdots_ratio <- ratio_calc(hypa_mdots) %>%
  filter(location_OT != "OT2")

V3_mdots_ratio <- ratio_calc(V3_mdots)

Z1_mdots_ratio <- ratio_calc(Z1_mdots)

## FANCFg2 + ZSCAN2g1 mdots

## filter data and split site column
temp_ratio <- filter(F2Z1_mdots, Reads_passing >= 1000 & # filters out less than 1000 reads
                       X_lab != "- - - - -") %>% # filters out no transfection control
  select(Experiment, Name, X_lab, Site, Indel_stringent) %>% # allows spread to work
  separate(col = Site, into = c("guide", "location"), sep = " (?=[^ ]+$)") # divide site into guide for join

# split into two temporary  files
ratio_on_f <- filter(temp_ratio, location == "ON" & guide == "FANCF sgRNA2")
ratio_off_f <- filter(temp_ratio, location != "ON" & guide == "FANCF sgRNA2") %>%
  mutate(Indel_stringent = replace(Indel_stringent, Indel_stringent == 0, 0.01))
ratio_on_z <- filter(temp_ratio, location == "ON" & guide == "ZSCAN2 sgRNA1")
ratio_off_z <- filter(temp_ratio, location != "ON" & guide == "ZSCAN2 sgRNA1") %>%
  mutate(Indel_stringent = replace(Indel_stringent, Indel_stringent == 0, 0.01))

# join based on guide, plot, calculate ratio
F2Z1_f_mdots_ratio <- full_join(ratio_on_f, ratio_off_f,
                                by = c("Experiment", "Name", "X_lab", "guide"),
                         suffix = c("_ON", "_OT")) %>%
  mutate(ratio = Indel_stringent_ON / Indel_stringent_OT) # calculates ratio

F2Z1_z_mdots_ratio <- full_join(ratio_on_z, ratio_off_z,
                                by = c("Experiment", "Name", "X_lab", "guide"),
                         suffix = c("_ON", "_OT")) %>%
  mutate(ratio = Indel_stringent_ON / Indel_stringent_OT) # calculates ratio

F2Z1_mdots_ratio <- bind_rows(F2Z1_f_mdots_ratio, F2Z1_z_mdots_ratio) %>%
  mutate(ratio = replace(ratio, ratio == 0, 0.1))

## plot VEGFAg2 mdots ratio
fig_s11a2 <- ggplot(data = V2_mdots_ratio,
                   aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 5.5, xmax = 6.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 7.5, xmax = 8.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = V2_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.8)) +
  scale_x_discrete(labels = addline_format(unique(V2_mdots_ratio$X_lab))) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(V2_mdots_ratio$location_OT),
                    labels = paste0(unique(V2_mdots_ratio$location_ON), ":",                                                 unique(V2_mdots_ratio$location_OT[!is.na(V2_mdots_ratio$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(V2_mdots_ratio$guide)) +
  scale_y_continuous(trans = "log2", expand = c(0, 0),
                     labels = c(1, 2, 4, 8, 16, 32, 64, 128),
                     breaks = c(1, 2, 4, 8, 16, 32, 64, 128)) +
  geom_blank(aes(y = 128))

## plot VEGFAg2 mdots with triplex only

fig_5a2 <- ggplot(data = V2_mdots_ratio_reduced,
                  aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  geom_beeswarm(cex = 2, size = 2.5, dodge.width = 0.3,
                pch = 21, alpha = 0.7) +
  stat_summary(data = V2_mdots_ratio_reduced, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.3)) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(V2_mdots_ratio_reduced$X_lab))) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(V2_mdots_ratio_reduced$location_OT),
                    labels = paste0(unique(V2_mdots_ratio_reduced$location_ON), ":",                                                 unique(V2_mdots_ratio_reduced$location_OT[!is.na(V2_mdots_ratio_reduced$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(V2_mdots_ratio_reduced$guide)) +
  scale_y_continuous(trans = "log2", expand = c(0, 0),
                     labels = c(1, 2, 4, 8, 16, 32, 64),
                     breaks = c(1, 2, 4, 8, 16, 32, 64)) +
  geom_blank(aes(y = 64))

## plot VEGFAg2 mdots with triplex only, ecas9

fig_5b2 <- ggplot(data = eCas9_mdots_ratio,
                  aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  geom_beeswarm(cex = 2, size = 2.5, dodge.width = 0.3,
                pch = 21, alpha = 0.7) +
  stat_summary(data = eCas9_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.3)) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(eCas9_mdots_ratio$X_lab))) +
  scale_y_continuous(trans = "log2",
                     expand = c(0, 0), limits = c(1, 256),
                     labels = c(1, 4, 16, 64, 256),
                     breaks = c(1, 4, 16, 64, 256)) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(eCas9_mdots_ratio$location_OT),
                    labels = paste0(unique(eCas9_mdots_ratio$location_ON), ":",                                                 unique(eCas9_mdots_ratio$location_OT[!is.na(eCas9_mdots_ratio$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(eCas9_mdots_ratio$guide))

## plot VEGFAg2 mdots with triplex only, hf1
fig_s12a2 <- ggplot(data = hf1_mdots_ratio,
                    aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  geom_beeswarm(cex = 1.2, size = 2, dodge.width = 0.3,
                pch = 21, alpha = 0.7) +
  stat_summary(data = hf1_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.3)) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(hf1_mdots_ratio$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 20),
                     breaks = seq(0, 20, by = 5)) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(hf1_mdots_ratio$location_OT),
                    labels = paste0(unique(hf1_mdots_ratio$location_ON), ":",                                                 unique(hf1_mdots_ratio$location_OT[!is.na(hf1_mdots_ratio$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(hf1_mdots_ratio$guide))

## plot VEGFAg2 mdots with triplex only, hypa
fig_s12b2 <- ggplot(data = hypa_mdots_ratio,
                    aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  geom_beeswarm(cex = 1.2, size = 2.5, dodge.width = 0.3,
                pch = 21, alpha = 0.7) +
  stat_summary(data = hypa_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.3)) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(hypa_mdots_ratio$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 10),
                     breaks = seq(0, 10, by = 2)) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(hypa_mdots_ratio$location_OT),
                    labels = paste0(unique(hypa_mdots_ratio$location_ON), ":",                                                 unique(hypa_mdots_ratio$location_OT[!is.na(hypa_mdots_ratio$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(hypa_mdots_ratio$guide))

## plot VEGFAg3 mdots ratio
fig_s11b2 <- ggplot(data = V3_mdots_ratio,
                   aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 5.5, xmax = 6.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 7.5, xmax = 8.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = V3_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.8)) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(V3_mdots_ratio$X_lab))) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(V3_mdots_ratio$location_OT),
                    labels = paste0(unique(V3_mdots_ratio$location_ON), ":",                                                 unique(V3_mdots_ratio$location_OT[!is.na(V3_mdots_ratio$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(V3_mdots_ratio$guide)) +
  scale_y_continuous(trans = "log2", expand = c(0, 0), limits = c(1, 128),
                     labels = c(1, 2, 4, 8, 16, 32, 64, 128),
                     breaks = c(1, 2, 4, 8, 16, 32, 64, 128))

## plot ZSCAN2g1 mdots ratio
fig_s11c2 <- ggplot(data = Z1_mdots_ratio,
                    aes(fct_inorder(X_lab), ratio, fill = location_OT)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = 1, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = 1, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                alpha = 0.7, pch = 21) +
  stat_summary(data = Z1_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.8)) + 
  scale_x_discrete(labels = addline_format(unique(Z1_mdots_ratio$X_lab))) +
  scale_fill_manual(values = pal[2:4],
                    breaks = unique(Z1_mdots_ratio$location_OT),
                    labels = paste0(unique(Z1_mdots_ratio$location_ON), ":",                                                 unique(Z1_mdots_ratio$location_OT[!is.na(Z1_mdots_ratio$location_OT)]))) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(Z1_mdots_ratio$guide)) +
  scale_y_log10(expand = c(0, 0), limits = c(1, 10000))

options(scipen = 100000)

## plot FANCFg2 + ZSCAN2g1 mdots ratio
fig_s11d2 <- ggplot(data = F2Z1_mdots_ratio,
                   aes(fct_inorder(X_lab), ratio,
                       fill = location_OT)) +
  annotate(geom = "rect", xmin = 1.5, xmax = 2.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 3.5, xmax = 4.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 5.5, xmax = 6.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 7.5, xmax = 8.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  annotate(geom = "rect", xmin = 9.5, xmax = 10.5,
           ymin = 0, ymax = Inf, alpha = 0.4, fill = "grey") +
  geom_beeswarm(size = 2.5, cex = 1.2, dodge.width = 0.8,
                pch = 21, alpha = 0.7) +
  stat_summary(data = F2Z1_mdots_ratio, fun = "mean",
              geom = "point", pch = "_", size = 8,
              position = position_dodge(0.8)) +
  scale_x_discrete(labels = addline_format(unique(F2Z1_mdots_ratio$X_lab))) +
  scale_fill_manual(values = c(pal[3], pal[2]),
                    breaks = unique(F2Z1_mdots_ratio$location_OT[!is.na(F2Z1_mdots_ratio$location_OT)]),
                    labels = paste0(unique(F2Z1_mdots_ratio$guide), " ON:",                                                 unique(F2Z1_mdots_ratio$location_OT[!is.na(F2Z1_mdots_ratio$location_OT)]))) +
  scale_color_manual(values = c("black", "black")) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0(unique(F2Z1_mdots_ratio$guide)[1], " and ", unique(F2Z1_mdots_ratio$guide)[2])) +
  scale_y_log10(expand = c(0, 0), breaks = c(0.1, 10, 1000, 100000),
                labels = c(0.1, 10, 1000, 100000)) + 
  geom_blank(aes(y = 100000)) +
  guides(fill = guide_legend(nrow = 1))

## fig labels
v2_labels <- label_plot(V2_mdots, aes(x = as_factor(Experiment),
                                      y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = "\nCas9\n\nVEGFA sgRNA2\n\nOT1 dRNA\n\nOT2 dRNA\n\nOT17 dRNA")

v2a_labels <- v2_labels + 
  labs(title = "a")

v2b_labels <- v2_labels + 
  labs(title = "b")

v3_labels <- label_plot(V3_mdots, aes(x = as_factor(Experiment),
                                      y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = "\nCas9\n\nVEGFA sgRNA3\n\nOT2 dRNA\n\nOT4 dRNA\n\nOT18 dRNA")

v3b_labels <- v3_labels +
  labs(title = "b")

z1_labels <- label_plot(Z1_mdots, aes(x = as_factor(Experiment),
                                      y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = "\nCas9\n\nZSCAN2 sgRNA1\n\nOT1 dRNA\n\nOT2 dRNA")

z1c_labels <- z1_labels +
  labs(title = "c")

f2z1_labels <- label_plot(F2Z1_mdots, aes(x = as_factor(Experiment),
                                      y = Indel_stringent)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = "\nCas9\n\nFANCF sgRNA2\n\nZSCAN2 sgRNA1\n\nFANCF OT1 dRNA\n\nZSCAN2 OT2 dRNA")

f2z1d_labels <- f2z1_labels +
  labs(title = "d")

# store and save plots as aligned grids
## fig 5 - mdots main VEGFA sgRNA2 
fig_5 <- v2a_labels + fig_5a1 + v2_labels + fig_5a2 + 
  v2b_labels + fig_5b1 + v2_labels + fig_5b2 +
  plot_layout(ncol = 4, widths = c(0.05, 1, 0.05, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

ggsave("Stat_analysis/Main_figures/Fig5_mdots.pdf", fig_5,
       height = 12, width = 12)

## fig s11 - mdots extended
fig_s11 <- v2a_labels + plot_spacer() + fig_s11a1 + v2_labels + fig_s11a2 +
  v3b_labels + plot_spacer() + fig_s11b1 + v3_labels + fig_s11b2 +
  z1c_labels + plot_spacer() + fig_s11c1 + z1_labels + fig_s11c2 +
  f2z1d_labels + plot_spacer() + fig_s11d1 + f2z1_labels + fig_s11d2 +
  plot_layout(ncol = 5, widths = c(0.05, 0.1, 1, 0.05, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

ggsave("Stat_analysis/Supp_figures/FigS11_mdots_extended.pdf", fig_s11,
       height = 24, width = 16)

fig_s12 <- v2a_labels + fig_s12a1 + v2_labels + fig_s12a2 + 
  v2b_labels + fig_s12b1 + v2_labels + fig_s12b2 +
  plot_layout(ncol = 4, widths = c(0.05, 1, 0.05, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

ggsave("Stat_analysis/Supp_figures/FigS12_hifi_mdots.pdf", fig_s12,
       height = 12, width = 12)

## create source file for fig 5a
fig_5a_indels_source <- V2_mdots_reduced %>%
  mutate(Panel = "a") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_5a_ratio_source <- V2_mdots_ratio_reduced %>%
  mutate(Panel = "a") %>%
  ratio_source()

fig_5a_source <- left_join(fig_5a_indels_source, fig_5a_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## create source file for fig 5b
fig_5b_indels_source <- eCas9_mdots %>%
  mutate(Panel = "b") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_5b_ratio_source <- eCas9_mdots_ratio %>%
  mutate(Panel = "b") %>%
  ratio_source()

fig_5b_source <- left_join(fig_5b_indels_source, fig_5b_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON, -Ratio_OT2) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## create source file for fig s11a
fig_s11a_indels_source <- V2_mdots %>%
  mutate(Panel = "a") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_s11a_ratio_source <- V2_mdots_ratio %>%
  mutate(Panel = "a") %>%
  ratio_source()

fig_s11a_source <- left_join(fig_s11a_indels_source, fig_s11a_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## create source file for fig s11b
fig_s11b_indels_source <- V3_mdots %>%
  mutate(Panel = "b") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_s11b_ratio_source <- V3_mdots_ratio %>%
  mutate(Panel = "b") %>%
  ratio_source()

fig_s11b_source <- left_join(fig_s11b_indels_source, fig_s11b_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## create source file for fig s11c
fig_s11c_indels_source <- Z1_mdots %>%
  mutate(Panel = "c") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_s11c_ratio_source <- Z1_mdots_ratio %>%
  mutate(Panel = "c") %>%
  ratio_source()

fig_s11c_source <- left_join(fig_s11c_indels_source, fig_s11c_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## create source file for fig s11d
fig_s11d_indels_source <- F2Z1_mdots %>%
  mutate(Panel = "d") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_s11d_ratio_source <- F2Z1_mdots_ratio %>%
  mutate(Panel = "d") %>%
  ratio_source()

fig_s11d_source <- left_join(fig_s11d_indels_source, fig_s11d_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = c("Guide", "Off-target"),
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-"Ratio_FANCF sgRNA2_ON", -"Ratio_ZSCAN2 sgRNA1_ON") %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace_all(., "_", " ")))

## create source file for fig s12a
fig_s12a_indels_source <- hf1_mdots %>%
  mutate(Panel = "a") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_s12a_ratio_source <- hf1_mdots_ratio %>%
  mutate(Panel = "a") %>%
  ratio_source()

fig_s12a_source <- left_join(fig_s12a_indels_source, fig_s12a_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON, -Ratio_OT2) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## create source file for fig s12b
fig_s12b_indels_source <- hypa_mdots %>%
  mutate(Panel = "b") %>%
  select(Panel, Name, guide, target, X_lab, Indel_stringent) %>%
  rename("Guide" = guide,
         "Off-target" = target,
         "X label" = X_lab)

fig_s12b_ratio_source <- hypa_mdots_ratio %>%
  mutate(Panel = "b") %>%
  ratio_source()

fig_s12b_source <- left_join(fig_s12b_indels_source, fig_s12b_ratio_source,
                           by = c("Panel", "Name", "Guide", "Off-target", "X label")) %>%
  pivot_wider(names_from = "Off-target",
              values_from = c("Indel_stringent", "Ratio")) %>%
  select(-Ratio_ON, -Ratio_OT2) %>%
  rename_at(vars(contains("Indel_stringent")), list(~gsub("Indel_stringent", "Indels ", .))) %>%
  rename_all(list(~str_replace(., "_", " ")))

## append source files
source_data_list <- append(source_data_list,
                           list("Fig5a" = fig_5a_source,
                                "Fig5b" = fig_5b_source,
                                "FigS11a" = fig_s11a_source,
                                "FigS11b" = fig_s11b_source,
                                "FigS11c" = fig_s11c_source,
                                "FigS11d" = fig_s11d_source,
                                "FigS12a" = fig_s12a_source,
                                "FigS12b" = fig_s12b_source))

```

```{r dRNA only controls}

## read dRNA only .csv file into memory, calculate t.test
dRNA_only <- read_csv("Data/dRNA_only_controls.csv") %>%
  filter(Reads_passing >= 1000 &
           Experiment != 5) %>%
  select(Experiment, Site, Indel_stringent, Name, X_lab) %>%
  group_by(Experiment, X_lab, Site) %>%
  summarise(Indel_stringent = list(Indel_stringent)) %>%
  spread(X_lab, Indel_stringent) %>%
  group_by(Experiment, Site) %>%
  mutate(n = (length(unlist(dRNA)) + length(unlist(NT))) / 2, 
         diff = mean(unlist(dRNA)) - mean(unlist(NT)),
         se = (1.96 * sqrt((sd(unlist(dRNA))^2 / length(unlist(dRNA))) + (sd(unlist(NT))^2) / length(unlist(NT)))), 
         se_max = diff + se,
         se_min = diff - se,
         p = t.test(unlist(dRNA), unlist(NT), alternative = "greater")$p.value) %>%
  ungroup()

# ===========================================================================================

## prepare data for fig s1c
dRNA_only2 <- read_csv("Data/dRNA_only_controls.csv") %>%
  filter(Experiment == 1 &
           Reads_passing >= 1000 &
           ## remove predicted sites since they are the same as original sites
           !grepl("predicted$", Site)) %>%
  mutate(X_lab = gsub("dRNA", "+ - +", X_lab),
         X_lab = gsub("NT", "- - -", X_lab),
         Site = gsub("g", " sgRNA", Site),
         guide = word(Site, start = 1, end = 2),
         target = word(Site, -1))

## make plot
fig_s1c <- ggplot(data = dRNA_only2,
                  aes(x = fct_inorder(X_lab),
                      y = Indel_stringent,
                      fill = Site)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21, alpha = 0.7) +
  stat_summary(fun = "mean", geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(dRNA_only2$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 0.025),
                     breaks = seq(0, 0.025, by = 0.005)) +
  scale_fill_manual(values = pal,
                     breaks = unique(dRNA_only2$Site),
                     labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(unique(dRNA_only2$Site)[2])

## create source file for fig s1c
fig_s1c_source <- dRNA_only2 %>%
  mutate(Panel = "c") %>%
  indel_source()

## append fig s1c to list
source_data_list <- append(source_data_list, list("FigS1c" = fig_s1c_source))

# ===========================================================================================

## draw fig 1a
fig_s1a <- ggdraw() +
  theme_cowplot() +
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  labs(title = "a") +
  draw_image(magick::image_read_pdf("Images/figs1a.pdf", density = 600))

## align fig s1
fig_s1_bottom <- label_s1b + plot_spacer() + fig_s1b1 +
  fig_s1_label + plot_spacer() + fig_s1b2 +
  label_s1c + plot_spacer() + fig_s1c +
  plot_spacer() + plot_spacer() + plot_spacer() +
  plot_layout(ncol = 6, widths = c(0.05, 0.05, 1))

fig_s1 <- (fig_s1a / fig_s1_bottom)  &
  theme(plot.margin = margin(t = 7, b = 20, r = 7, l = 7, unit = "pt"))

ggsave("Stat_analysis/Supp_figures/FigS1_FANCF_controls.pdf", fig_s1,
       height = 18, width = 12)

# ===========================================================================================

# takes values with equal variance (which t.test throws as NaN for)
# and sets p to 1 (since they are not in actuality different)
dRNA_only[is.na(dRNA_only)] <- 1

dRNA_only <- dRNA_only %>%
  mutate(padj = p.adjust(p, "bonferroni"))

# split to re-join for easier comprehension
d1 <- filter(dRNA_only, grepl("N$", dRNA_only$Site)) %>%
  rename(p_ON = p) %>%
  rename(p_adj_ON = padj) %>%
  rename(n_ON = n) %>%
  rename(diff_ON = diff)
d2 <- filter(dRNA_only, grepl("[0-9]$|T$", dRNA_only$Site)) %>%
  rename(p_OT = p) %>%
  rename(p_adj_OT = padj) %>%
  rename(n_OT = n) %>%
  rename(diff_OT = diff)
d3 <- filter(dRNA_only, grepl("N\\spredicted$", dRNA_only$Site)) %>%
  rename(p_ON_pred = p) %>%
  rename(p_adj_ON_pred = padj) %>%
  rename(n_ON_pred = n) %>%
  rename(diff_ON_pred = diff)
d4 <- filter(dRNA_only, grepl("[0-9]\\spredicted$|T\\spredicted$", dRNA_only$Site)) %>%
  rename(p_OT_pred = p) %>%
  rename(p_adj_OT_pred = padj) %>%
  rename(n_OT_pred = n) %>%
  rename(diff_OT_pred = diff)

# join
dRNA_only_table <- full_join(d1, d2, by = "Experiment") %>%
  full_join(d3, by = "Experiment") %>%
  full_join(d4, by = "Experiment")

# select order for printing
dRNA_only_table_s2 <- select(dRNA_only_table, Site.y,
                             ends_with("ON"), ends_with("OT"), -starts_with("n"))

dRNA_only_table_s3 <- select(dRNA_only_table, Site.y,
                             ends_with("ON_pred"), ends_with("OT_pred"), -starts_with("n"))

## output table s2
write_csv(dRNA_only_table_s2, "Stat_analysis/Supp_tables/TableS2_dRNA_only_controls.csv")

write_csv(dRNA_only_table_s3, "Stat_analysis/Supp_tables/TableS3_dRNA_only_controls_predicted_sites.csv")

# output table
kable(dRNA_only_table) %>%
  kable_styling(latex_options="scale_down")

```

```{r nontargeting controls}

## read in nontargeting dRNA data and create list of plots
nontargeting <- read_csv("Data/nontargeting_controls.csv") %>%
  filter(Reads_passing > 1000 &
         Experiment != 12 &
         Experiment != 1) %>%
  mutate(Site = gsub("g", " sgRNA", Site),
         X_lab = gsub("[0-9]", "+", X_lab),
         X_lab = gsub("s", "", X_lab),
         guide = word(Site, start = 1L, end = 2L),
         target = word(Site, start = 3L)) %>%
  group_by(Experiment) %>%
  arrange(Site) %>%
  ungroup() %>%
  mutate(Experiment = factor(Experiment, unique(Experiment))) %>%
  arrange(Experiment) %>%
  group_by(Experiment) %>%
  mutate(Panel = letters[Experiment]) %>%
  nest() %>%
  ## make indel plot for nontargeting samples
  mutate(plot = map2(data, Experiment, ~ggplot(data = .x) +
                       aes(x = fct_inorder(X_lab), Indel_stringent, fill = Site) +
                       geom_beeswarm(size = 2.5, cex = 2, pch = 21, alpha = 0.7) +
                       stat_summary(data = .x, fun = "mean",
                                    geom = "point", pch = "_", size = 8) + # plots mean of each site
                       scale_x_discrete(labels = addline_format(unique(.$X_lab))) +
                       scale_y_continuous(expand = c(0, 0),
                                          limits = c(0, round_max(.)),
                                          breaks = seq(0, round_max(.), by = (round_max(.) / 5))) +
                       scale_fill_manual(values = pal,
                                          breaks = unique(.$Site),
                                          labels = c("on-target", "off-target")) +
                       ylab("Indel frequency (%)") +
                       ggtitle(unique(.$Site)[2])),
         label_plots = map2(data, Experiment,
                           ~label_plot(data = .x) +
                             aes(Date,
                                 Indel_stringent) +
                             geom_point(color = "NA") +
                             labs(title = unique(.$Panel)) +
                             scale_x_discrete(labels = label_grab(.)))) %>%
  ## join to plot side by side
  pivot_longer(cols = c("label_plots", "plot"),
               names_to = "plottype",
               values_to = "plot")

## arrange nontargeting dRNA plots
fig_s5 <- wrap_plots(nontargeting$plot, ncol = 6,
                     widths = c(0.25, 1, 0.25, 1, 0.25, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 25, l = 7, "pt"))

## save Figure S5 - nontargeting dRNA controls
ggsave("Stat_analysis/Supp_figures/FigS5_nontargeting.pdf", fig_s5,
       height = 24, width = 18)

## create source file for fig s5
fig_s5_source <- nontargeting %>%
  select(Experiment, data) %>%
  unique() %>%
  unnest(data) %>%
  mutate(X_lab = factor(X_lab, levels = c("+ + -", "+ + +", "+ + dNT", "- - -"))) %>%
  indel_source() %>%
  arrange(Panel, `X label`)

## split by panel
fig_s5_list <- fig_s5_source %>%
  named_group_split(Panel)

## change names to figure panels
names(fig_s5_list) <- paste0("FigS5", names(fig_s5_list))

## append fig s5
source_data_list <- append(source_data_list, fig_s5_list)

# ===========================================================================================

nontargeting_FANCF <- read_csv("Data/nontargeting_controls.csv") %>%
  filter(Reads_passing > 1000 &
         Experiment == 1) %>%
  mutate(X_lab = gsub("[0-9]", "\\+", X_lab),
         Site = gsub("g", " sgRNA", Site),
         guide = word(Site, start = 1, end = 2))

## plot fig 1b - indels
fig_1b1 <- ggplot(data = nontargeting_FANCF,
                  aes(fct_inorder(X_lab), Indel_stringent,
                      fill = Site)) +
  geom_beeswarm(size = 2.5, cex = 2,
                pch = 21, alpha = 0.7) +
  stat_summary(data = nontargeting_FANCF, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(nontargeting_FANCF$X_lab))) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 30),
                     breaks = seq(0, 30, by = 10)) +
  scale_fill_manual(values = pal,
                    breaks = unique(nontargeting_FANCF$Site),
                    labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(unique(nontargeting_FANCF$Site)[2])

## take ratio
ratio_nontargeting <- ratio_calc(nontargeting_FANCF)

## plot fig 1b - ratio
fig_1b2 <- ggplot(data = ratio_nontargeting,
                  aes(fct_inorder(X_lab), ratio)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21,
                alpha = 0.7, fill = pal[5]) +
  stat_summary(data = ratio_nontargeting, fun = "mean",
               geom = "point", pch = "_", size = 8) + 
  scale_x_discrete(labels = addline_format(unique(ratio_nontargeting$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(ratio_nontargeting)),
                     breaks = seq(0, round_max(ratio_nontargeting), by = 10)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(unique(nontargeting_FANCF$Site)[2])

fig_1_label <- label_plot(ratio_nontargeting,
                           aes(x = guide,
                               y = ratio)) +
  geom_point(color = NA) + 
  scale_x_discrete(labels = label_grab(ratio_nontargeting))

fig_1b_label <- fig_1_label +
  labs(title = "b")

fig_1c_label <- label_plot(ratio_nontargeting,
                           aes(x = guide,
                               y = ratio)) +
  geom_point(color = NA) +
  theme(axis.text.x = element_blank()) +
  labs(title = "c")

## create source data file for fig 1b
fig_1b_source <- nontargeting_FANCF %>%
  mutate(target = word(Site, -1),
         Panel = "b",
         X_lab = factor(X_lab, levels = c("+ + -", "+ + +", "+ + dNT", "- - -"))) %>%
  indel_source() %>%
  arrange(`X label`) %>%
  mutate(Ratio = ifelse(`X label` == "- - -", NA, `Indels ON` / `Indels OT`))

## append to source data file
source_data_list <- append(source_data_list, list("Fig1b" = fig_1b_source))

```

```{r U2OS}

## read in data from U2OS cells
u2os <- read_csv("Data/U2OS.csv") %>%
  filter(Reads_passing >= 1000) %>%
  mutate(Site = gsub("g", " sgRNA", Site),
         X_lab = paste0(X_lab, "\n"),
         X_lab = gsub("[0-9]", "+", X_lab),
         guide = word(Site, start = 1, end = 2),
         target = word(Site, -1))

## create ratio for U2OS cells
u2os_ratio <- u2os %>%
  filter(word(X_lab, start = 1) != "-") %>%
  ratio_calc() %>%
  group_by(X_lab, Experiment) %>%
  mutate(avg = mean(ratio, na.rm = TRUE),
         n_ON = sum(!is.na(Indel_stringent_ON)),
         n_OT = sum(!is.na(Indel_stringent_OT)),
         avg_ON = mean(Indel_stringent_ON, na.rm = TRUE),
         avg_OT = mean(Indel_stringent_OT, na.rm = TRUE),
         sem_ON = sd(Indel_stringent_ON, na.rm = TRUE) / sqrt(n_ON),
         sem_OT = sd(Indel_stringent_OT, na.rm = TRUE) / sqrt(n_OT),
         sem_ratio = (avg_ON / avg_OT) * sqrt((sem_ON / avg_ON)^2 + (sem_OT / avg_OT)^2)) %>%
  ungroup()

# =============================================================================================

## filter for FANCF experiment
u2os_f <- u2os %>%
  filter(Experiment == 1)

u2os_ratio_f <- u2os_ratio %>%
  filter(Experiment == 1)

## plot indel data - U2OS
fig_1d1 <- ggplot(data = u2os_f,
                  aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  #geom_blank(aes(y = ymin)) +
  #geom_blank(aes(y = ymax)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21, alpha = 0.7) +
  stat_summary(data = u2os_f, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(u2os_f$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 60), breaks = seq(0, 60, by = 10)) +
  scale_fill_manual(values = pal, breaks = unique(u2os_f$Site),
                     labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(paste0("U2OS - ", unique(u2os_f$Site)[2]))

## plot ratios - U2OS
fig_1d2 <- ggplot(data = u2os_ratio_f,
                  aes(fct_inorder(X_lab), ratio)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21, alpha = 0.7, fill = pal[5]) +
  stat_summary(data = u2os_ratio_f, fun = "mean",
               geom = "point", pch = "_", size = 8) + # plots mean of each site
  scale_x_discrete(labels = addline_format(unique(u2os_ratio_f$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(u2os_ratio_f)),
                     breaks = seq(0, round_max(u2os_ratio_f), by = 50)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0("U2OS - ", unique(u2os_ratio_f$guide), " ",
                 unique(u2os_ratio_f$location_OT)))

## create label for fig 1d and s6a
fig_1d_label <- label_plot(data = u2os_f,
                           aes(x = as_factor(Experiment),
                               y = Indel_stringent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\nCas9\n\nFANCF sgRNA2\n\ndRNA")

fig_1d1_label <- fig_1d_label +
  labs(title = "d")

## create source file for fig 1d
fig_1d_indel_source <- u2os_f %>%
  mutate(Panel = "d") %>%
  indel_source()

fig_1d_ratio_source <- u2os_ratio_f %>%
  mutate(Panel = "d") %>%
  ratio_source()

fig_1d_source <- left_join(fig_1d_indel_source, fig_1d_ratio_source)

## append to source data list
source_data_list <- append(source_data_list, list("Fig1d" = fig_1d_source))

# =============================================================================================

## filter U2OS for VEGFA sgRNA3
u2os_v <- u2os %>%
  filter(Experiment == 2)

u2os_ratio_v <- u2os_ratio %>%
  filter(Experiment == 2)

## plot fig s6 - alternative site for U2OS
fig_s6a1 <- ggplot(data = u2os_v,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = u2os_v, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(u2os_v$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(u2os_v)),
                     breaks = seq(0, round_max(u2os_v), by = round_max(u2os_v) / 4)) +
  scale_fill_manual(values = pal, breaks = unique(u2os_v$Site),
                     labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(paste0("U2OS - ", unique(u2os_v$Site)[2]))
  
## plot ratios
fig_s6a2 <- ggplot(data = u2os_ratio_v,
                    aes(fct_inorder(X_lab), ratio)) +
  geom_beeswarm(size = 2.5, cex = 1.2, pch = 21,
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = u2os_ratio_v, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(u2os_ratio_v$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(u2os_ratio_v)),
                     breaks = seq(0, round_max(u2os_ratio_v), by = round_max(u2os_ratio_v) / 5)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0("U2OS - ", unique(u2os_ratio_v$guide), " ",
                 unique(u2os_ratio_v$location_OT)))

## create label for fig s6a
fig_s6a_label <- label_plot(data = u2os_v,
                            aes(x = as_factor(Experiment),
                                y = Indel_stringent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\nCas9\n\nVEGFA sgRNA3\n\ndRNA")

fig_s6a1_label <- fig_s6a_label +
  labs(title = "a")

## create source file for fig s6a
fig_s6a_indel_source <- u2os_v %>%
  mutate(Panel = "a") %>%
  indel_source()

fig_s6a_ratio_source <- u2os_ratio_v %>%
  mutate(Panel = "a") %>%
  ratio_source()

fig_s6a_source <- left_join(fig_s6a_indel_source, fig_s6a_ratio_source)

## append to source data list
source_data_list <- append(source_data_list, list("FigS6a" = fig_s6a_source))

```

```{r ELF}

## read in data from ELF cells
elf <- read_csv("Data/elf.csv") %>%
  filter(Reads_passing >= 1000) %>%
  mutate(guide = word(Site, 1),
         Site = gsub("g", " sgRNA", Site),
         guide = gsub("g", " sgRNA", guide),
         target = word(Site, -1),
         X_lab = gsub("[0-9]", "+", X_lab))

## calculate ratios for ELF 
elf_ratio <- filter(elf, X_lab != "- - - +") %>%
  ratio_calc() %>%
  group_by(X_lab, Experiment) %>%
  mutate(avg = mean(ratio, na.rm = TRUE),
         n_ON = sum(!is.na(Indel_stringent_ON)),
         n_OT = sum(!is.na(Indel_stringent_OT)),
         avg_ON = mean(Indel_stringent_ON, na.rm = TRUE),
         avg_OT = mean(Indel_stringent_OT, na.rm = TRUE),
         sem_ON = sd(Indel_stringent_ON, na.rm = TRUE) / sqrt(n_ON),
         sem_OT = sd(Indel_stringent_OT, na.rm = TRUE) / sqrt(n_OT),
         sem_ratio = (avg_ON / avg_OT) * sqrt((sem_ON / avg_ON)^2 + (sem_OT / avg_OT)^2)) %>%
  ungroup()

# =============================================================================================

## filter ELF to FANCF sgRNA2
elf_f <- elf %>%
  filter(Experiment == 1)

elf_ratio_f <- elf_ratio %>% 
  filter(Experiment == 1)

## grab distinct labels (two experiments with negative controls)
labels_f <- elf_f %>%
  group_by(Date) %>%
  distinct(X_lab)

## plot fig 1e, panel 1 - indel for ELF
fig_1e1 <- ggplot(data = elf_f,
                  aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(cex = 2, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = elf_f, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  facet_grid(~ fct_inorder(Date), scale = "free", space = "free") +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank()) +
  scale_x_discrete(breaks = labels_f$X_lab,
                   labels = addline_format(labels_f$X_lab)) +
  scale_fill_manual(values = pal, breaks = unique(elf_f$Site),
                     labels = c("on-target", "off-target")) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(elf_f)),
                     breaks = seq(0, round_max(elf_f),
                                  by = round_max(elf_f) / 4)) +
  ylab("Indel frequency (%)") +
  ggtitle(paste0("ELF1 - ", unique(elf_f$Site)[2]))

## plot ratios 
fig_1e2 <- ggplot(data = elf_ratio_f,
                  aes(fct_inorder(X_lab), ratio)) +
  geom_beeswarm(size = 2.5, cex = 2, pch = 21,
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = elf_ratio_f, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(elf_ratio_f$X_lab))) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, round_max(elf_ratio_f)),
                     breaks = seq(0, round_max(elf_ratio_f),
                                  by = round_max(elf_ratio_f) / 5)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0("ELF1 - ", unique(elf_ratio_f$guide), " ",
                 unique(elf_ratio_f$location_OT)))

## create label for fig 1 and 5s
fig_1e_label <- label_plot(data = elf_f,
                           aes(x = as_factor(Experiment),
                               y = Indel_stringent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\niCas9\n\nFANCF sgRNA2\n\ndRNA\n\nDox")

fig_1e1_label <- fig_1e_label +
  labs(title = "e")

## plot fig 1a - dots schematic
fig_1a <- ggdraw() +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  labs(title = "a") +
  draw_image(magick::image_read_pdf("Images/fig1a.pdf", density = 600))

## make layout
layout_fig1 <- "
ABBBBCDDDD
EFFFFFFFFF
GHHHHIJJJJ
KLLLLMNNNN
"

## align plots
fig_1_bottom <- fig_1b_label + fig_1b1 + fig_1_label + fig_1b2 +
  fig_1c_label + fig_1c + 
  fig_1d1_label + fig_1d1 + fig_1d_label + fig_1d2 + 
  fig_1e1_label + fig_1e1 + fig_1e_label + fig_1e2 + 
  plot_layout(design = layout_fig1)

## align fig 1a 
fig_1 <- fig_1a / fig_1_bottom + plot_layout(heights = c(1, 2)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

## save fig 1
ggsave("Stat_analysis/Main_figures/Fig1_FANCF.pdf", fig_1,
       height = 30, width = 10)

## create source file for fig 1e
fig_1e_indel_source <- elf_f %>%
  mutate(Panel = "e") %>%
  indel_source()

fig_1e_ratio_source <- elf_ratio_f %>%
  mutate(Panel = "e") %>%
  ratio_source()

fig_1e_source <- left_join(fig_1e_indel_source, fig_1e_ratio_source)

## append to source data list
source_data_list <- append(source_data_list, list("Fig1e" = fig_1e_source))

# =============================================================================================

## filter HBB R03 for ELF
elf_h <- elf %>%
  filter(Experiment == 2)

## calculate ratios
elf_ratio_h <- elf_ratio %>%
  filter(Experiment == 2)

## add labels
labels_h <- elf_h %>%
  group_by(Date) %>%
  distinct(X_lab)

# =============================================================================================

## plot indels for HBB R03 - ELF
fig_s6b1 <- ggplot(data = elf_h,
                    aes(fct_inorder(X_lab), Indel_stringent, fill = Site)) +
  geom_beeswarm(size = 2.5, cex = 1.2, pch = 21, alpha = 0.7) +
  stat_summary(data = elf_h, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  facet_grid(~ fct_inorder(Date), scale = "free", space = "free") +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank()) +
  scale_x_discrete(breaks = labels_h$X_lab,
                   labels = addline_format(labels_h$X_lab)) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(elf_h)),
                     breaks = seq(0, round_max(elf_h), by = 10)) +
  scale_fill_manual(values = pal, breaks = unique(elf_h$Site),
                     labels = c("on-target", "off-target")) +
  ylab("Indel frequency (%)") +
  ggtitle(paste0("ELF1 - ", unique(elf_h$Site)[2]))

## plot ratios
fig_s6b2 <- ggplot(data = elf_ratio_h,
                    aes(fct_inorder(X_lab), ratio)) +
  geom_beeswarm(size = 2.5, cex = 1.2, pch = 21,
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = elf_ratio_h, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(labels = addline_format(unique(elf_ratio_h$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, round_max(elf_ratio_h)),
                     breaks = seq(0, round_max(elf_ratio_h), by = 5)) +
  ylab("On-target/Off-target\nindel frequency ratio") +
  ggtitle(paste0("ELF1 - ", unique(elf_ratio_h$guide), " ", unique(elf_ratio_h$location_OT)))

## create label for fig s6b
fig_s6b_label <- label_plot(data = elf_h,
                            aes(x = as_factor(Experiment),
                                y = Indel_stringent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\niCas9\n\nHBB-R03\n\ndRNA\n\nDox")

fig_s6b1_label <- fig_s6b_label +
  labs(title = "b")

## align plots
fig_s6 <- fig_s6a1_label + plot_spacer() + fig_s6a1 +
  fig_s6a_label + fig_s6a2 +
  fig_s6b1_label + plot_spacer() + fig_s6b1 +
  fig_s6b_label + fig_s6b2 +
  plot_layout(ncol = 5, widths = c(0.05, 0.05, 1, 0.01, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 15, l = 7, unit = "pt"))

## save fig s6 - additional sites for other cell types
ggsave("Stat_analysis/Supp_figures/FigS6_additional_cell_types.pdf", fig_s6,
       height = 12, width = 12)

## create source file for fig s6b
fig_s6b_indel_source <- elf_h %>%
  mutate(Panel = "b") %>%
  indel_source()

fig_s6b_ratio_source <- elf_ratio_h %>%
  mutate(Panel = "b") %>%
  ratio_source()

fig_s6b_source <- left_join(fig_s6b_indel_source, fig_s6b_ratio_source)

## append to source data list
source_data_list <- append(source_data_list, list("FigS6b" = fig_s6b_source))

```

```{r BFP dRECS}

## read in bfp/gfp flow summary data
bfp <- read_csv("Data/drecs.csv") %>%
  mutate(X_lab = fct_relevel(X_lab, "- - - -", after = Inf)) %>%
  rename(Indels = NHEJ)

## filter for dRNA screen
bfp_screen <- bfp %>%
  filter(Experiment == 1 &
           Days == 4 &
           word(X_lab, start = 2) != "2") %>%
  select(-Ratio_dRNA, -Order) %>%
  pivot_longer(cols = WT:HDR_Percent_Total,
               names_to = "Editing",
               values_to = "Percent") %>%
  mutate(facet_group = ifelse(Percent > 5, 1, 2),
         sgRNA_group = word(X_lab, start = 2, end = 3)) %>%
  group_by(X_lab, Editing) %>%
  mutate(avg = mean(Percent)) %>%
  ungroup() %>%
  group_by(sgRNA_group, Editing) %>%
  mutate(fold_change = Percent / first(avg)) %>%
  ungroup()
  
## label for fig s13b
fig_s13bd_label <- label_plot(data = bfp_screen,
                             aes(x = as_factor(Experiment),
                                 y = Percent)) +
  geom_point(color = NA) +
  scale_x_discrete(labels = "\nCas9\n\nBFP sgRNA\n\nDonor\n\ndRNA")

fig_s13b_label <- fig_s13bd_label +
  labs(title = "b")

## dRNA screen - indels and HDR
bfp_screen_s13b1 <- bfp_screen %>%
  filter(Editing == "HDR" | Editing == "Indels") %>%
  mutate(sgRNA_group = word(X_lab, start = 2),
         sgRNA_group = factor(sgRNA_group, levels = c("1", "2", "-")),
         X_lab = gsub("^\\+ [0-9]", "+ +", X_lab),
         X_lab = factor(X_lab, levels = c("+ + - -", "+ + + -", "+ + + 1",
                                          "+ + + 2", "+ + + 3", "- - - -")),
         facet_group = ifelse(Percent > 5, 1, 2),
         ymin = ifelse(facet_group == 1, 10, 0),
         ymax = ifelse(facet_group == 1, 25, 3))

## plot fig s13b panel 1, dRNA screen - indels and HDR
fig_s13b1 <- ggplot(data = bfp_screen_s13b1,
                    aes(X_lab, Percent, fill = Editing)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = bfp_screen_s13b1, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  facet_grid(facet_group ~ ., scales = "free", space = "free_x") +
  theme(strip.background = element_blank(),
        strip.text = element_blank(),
        panel.spacing = unit(15, "pt")) +
  scale_x_discrete(breaks = unique(bfp_screen_s13b1$X_lab),
                   labels = addline_format(unique(bfp_screen_s13b1$X_lab))) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = pal, breaks = unique(bfp_screen_s13b1$Editing),
                     labels = unique(gsub(".*\\s", "",
                                          unique(bfp_screen_s13b1$Editing)))) +
  ylab("% editing") +
  geom_blank(aes(y = ymin)) +
  geom_blank(aes(y = ymax))

# =============================================================================================

## filter for dRNA screen - % HDR of all cutting events
bfp_screen_s13b2 <- bfp_screen %>%
  filter(Editing == "HDR_Percent_Total" &
           word(X_lab, start = 3) != "-") %>%
  mutate(sgRNA_group = word(X_lab, start = 2),
         X_lab = gsub("^\\+ [0-9]", "+ +", X_lab))

## plot fig s13b, panel 2 - % HDR of all cutting events 
fig_s13b2 <- ggplot(data = bfp_screen_s13b2, aes(X_lab, Percent)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21,
                alpha = 0.7, fill = pal[5]) +
  stat_summary(data = bfp_screen_s13b2, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(breaks = unique(bfp_screen_s13b2$X_lab),
                   labels = addline_format(unique(bfp_screen_s13b2$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 5),
                     breaks = seq(0, 5, by = 1)) +
  ylab("% HDR of total cutting")

# =============================================================================================

## filter for dRNA screen - fold change in HDR
bfp_screen_s13b3 <- bfp_screen %>%
  filter(sgRNA_group == "1 +" |
         sgRNA_group == "2 +") %>%
  filter(Editing == "HDR") %>%
  mutate(X_lab = gsub("^\\+ [0-9]", "+ +", X_lab))

## plot fig s13b, panel 3 - fold change in HDR
fig_s13b3 <- ggplot(data = bfp_screen_s13b3, aes(X_lab, fold_change)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21,
                alpha = 0.7, fill = pal[3]) +
  stat_summary(data = bfp_screen_s13b3, fun = "mean",
               geom = "point", pch = "_", size = 8) + # plots mean of each site
  scale_x_discrete(breaks = unique(bfp_screen_s13b3$X_lab),
                   labels = addline_format(unique(bfp_screen_s13b3$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 4),
                     breaks = seq(0, 4, by = 1)) +
  ylab("Fold change in HDR")

## create source file for fig s13b
fig_s13b_source <- full_join(bfp_screen_s13b1, bfp_screen_s13b2) %>%
  mutate(Panel = "b") %>%
  select(Panel, Sample, X_lab, Editing, Percent, fold_change) %>%
  pivot_wider(names_from = Editing,
              values_from = c(Percent, fold_change)) %>%
  select(-c("fold_change_Indels", "fold_change_HDR_Percent_Total")) %>%
  mutate(fold_change_HDR = ifelse(X_lab == "+ + - -", NA,
                                  ifelse(X_lab == "- - - -", NA, fold_change_HDR))) %>%
  rename_all(list(~str_replace_all(., "_", " "))) %>%
  rename_at(vars(contains("Percent ")), list(~gsub("Percent ", "", .))) %>%
  rename("X label" = `X lab`,
         "Fold change in HDR" = `fold change HDR`,
         "Percent HDR of total editing" = `HDR Total`)

## append to source list
source_data_list <- append(source_data_list, list("FigS13b" = fig_s13b_source))

# =============================================================================================

## filter for dRNA ratio
bfp_dRNA_amount <- bfp %>%
  filter(Experiment == 2 &
         !is.na(X_lab) &
         word(X_lab, start = 2) != "2") %>%
  select(-Order) %>%
  pivot_longer(cols = WT:HDR_Percent_Total,
               names_to = "Editing",
               values_to = "Percent") %>%
  mutate(sgRNA_group = word(X_lab, start = 2, end = 3),
         new_X = paste0(X_lab, " ", Ratio_dRNA)) %>%
  group_by(X_lab, Editing) %>%
  mutate(avg = mean(Percent)) %>%
  ungroup() %>%
  group_by(sgRNA_group, Editing) %>%
  mutate(fold_change = Percent / first(avg)) %>%
  ungroup()

## filter for dRNA ratio
bfp_dRNA_amount_s13c1 <- bfp_dRNA_amount %>%
  filter(Editing == "HDR" | Editing == "Indels") %>%
  mutate(sgRNA_group = word(X_lab, start = 2),
         new_X = gsub("^\\+ [0-9]", "+ +", new_X),
         new_X = gsub("3", "+", new_X),
         facet_group = if_else(Percent > 5, 1, 2),
         ymin = if_else(facet_group == 1, 15, 0),
         ymax = if_else(facet_group == 1, 35, 3))

## plot fig s13c, panel 1 - indels and HDR
fig_s13c1 <- ggplot(data = bfp_dRNA_amount_s13c1, aes(new_X, Percent, fill = Editing)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = bfp_dRNA_amount_s13c1, fun = "mean",
               geom = "point", pch = "_", size = 8) + # plots mean of each site
  scale_x_discrete(breaks = unique(bfp_dRNA_amount_s13c1$new_X),
                   labels = addline_format(unique(bfp_dRNA_amount_s13c1$new_X))) +
  facet_grid(facet_group ~ ., scales = "free") +
  theme(strip.background = element_blank(),
        strip.text = element_blank(),
        panel.spacing = unit(15, "pt")) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = pal, breaks = unique(bfp_dRNA_amount_s13c1$Editing),
                     labels = unique(gsub(".*\\s", "",
                                          unique(bfp_dRNA_amount_s13c1$Editing)))) +
  ylab("% editing") +
  geom_blank(aes(y = ymin)) +
  geom_blank(aes(y = ymax))

## label for fig s13c
fig_s13c_label <- label_plot(data = bfp_dRNA_amount,
                             aes(x = as_factor(Experiment),
                                 y = Percent)) +
  geom_point(color = NA) +
  scale_x_discrete(labels = "\nCas9\n\nBFP sgRNA\n\nDonor\n\ndRNA\n\nDonor:dRNA ratio")

fig_s13c1_label <- fig_s13c_label +
  labs(title = "c")

# =============================================================================================

## filter dRNA ratio for % HDR of total cutting
bfp_dRNA_amount_s13c2 <- bfp_dRNA_amount %>%
  filter(Editing == "HDR_Percent_Total" &
                               new_X != "+ 1 - - 0" &
                               new_X != "+ 2 - - 0") %>%
  mutate(sgRNA_group = word(X_lab, start = 2),
         new_X = gsub("^\\+ [0-9]", "+ +", new_X),
         new_X = gsub("3", "+", new_X))

## plot fig s13c, panel 2 - % HDR of total cutting
fig_s13c2 <- ggplot(data = bfp_dRNA_amount_s13c2,
                    aes(new_X, Percent)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21,
                alpha = 0.7, fill = pal[5]) +
  stat_summary(data = bfp_dRNA_amount_s13c2, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(breaks = unique(bfp_dRNA_amount_s13c2$new_X),
                   labels = addline_format(unique(bfp_dRNA_amount_s13c2$new_X))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 10), breaks = seq(0, 10, by = 2),
                     labels = seq(0, 10, by = 2)) +
  ylab("% HDR of total editing")

# =============================================================================================

## filter dRNA ratio - fold change in HDR
bfp_dRNA_amount_s13c3 <- bfp_dRNA_amount %>%
  filter(sgRNA_group == "1 +" |
         sgRNA_group == "2 +") %>%
  filter(Editing == "HDR") %>%
  mutate(new_X = gsub("^\\+ [0-9]", "+ +", new_X),
         new_X = gsub("3", "+", new_X))

## plot fig 12c, panel 3 - fold change in HDR
fig_s13c3 <- ggplot(data = bfp_dRNA_amount_s13c3,
                   aes(new_X, fold_change)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21,
                alpha = 0.7, fill = pal[3]) +
  stat_summary(data = bfp_dRNA_amount_s13c3, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(breaks = unique(bfp_dRNA_amount_s13c3$new_X),
                   labels = addline_format(unique(bfp_dRNA_amount_s13c3$new_X))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 4)) +
  ylab("Fold change in HDR")

## create source file for fig s13c
fig_s13c_source <- full_join(bfp_dRNA_amount_s13c1, bfp_dRNA_amount_s13c2) %>%
  mutate(Panel = "c") %>%
  select(Panel, Ratio_dRNA, Sample, new_X, Editing, Percent, fold_change) %>%
  pivot_wider(names_from = Editing,
              values_from = c(Percent, fold_change)) %>%
  select(-c("fold_change_Indels", "fold_change_HDR_Percent_Total")) %>%
  mutate(fold_change_HDR = ifelse(new_X == "+ + - - 0", NA, fold_change_HDR)) %>%
  rename_all(list(~str_replace_all(., "_", " "))) %>%
  rename_at(vars(contains("Percent ")), list(~gsub("Percent ", "", .))) %>%
  rename("X label" = `new X`,
         "Fold change in HDR" = `fold change HDR`,
         "Percent HDR of total editing" = `HDR Total`,
         "dRNA:sgRNA ratio" = `Ratio dRNA`)

## append to source list
source_data_list <- append(source_data_list, list("FigS13c" = fig_s13c_source))

# =============================================================================================

## filter for testing different donor DNA types
bfp_donor_screen <- filter(bfp, Experiment == 3 &
                           Days == 4 &
                           word(X_lab, start = 3) != "block") %>%
  select(-Ratio_dRNA, -Order) %>%
  pivot_longer(cols = WT:HDR_Percent_Total,
               names_to = "Editing",
               values_to = "Percent") %>%
  mutate(X_lab = gsub("[0-9]", "+", X_lab),
         X_lab = factor(X_lab, levels = c("+ + - -", "+ + sym -",
                                          "+ + sym +", "+ + asym -",
                                          "+ + asym +", "- - - -")),
         facet_group = if_else(Percent > 10, 1, 2),
         ymin = if_else(facet_group == 1, 10, 0),
         ymax = if_else(facet_group == 1, 30, 5))

## filter donor screen for indels and HDR
bfp_donor_screen_s13d1 <- bfp_donor_screen %>%
  filter(Editing == "HDR" | Editing == "Indels")

## plot fig s13d, panel 1 - indels and HDR
fig_s13d1 <- ggplot(data = bfp_donor_screen_s13d1,
                    aes(X_lab, Percent, fill = Editing)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = bfp_donor_screen_s13d1, fun = "mean",
               geom = "point", pch = "_", size = 8) + 
  facet_grid(facet_group ~ ., scales = "free") +
  theme(strip.background = element_blank(),
        strip.text = element_blank(),
        panel.spacing = unit(15, "pt")) +
  scale_x_discrete(breaks = unique(bfp_donor_screen_s13d1$X_lab),
                   labels = addline_format(unique(bfp_donor_screen_s13d1$X_lab))) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = pal, breaks = unique(bfp_donor_screen_s13d1$Editing),
                     labels = unique(gsub(".*\\s", "", unique(bfp_donor_screen_s13d1$Editing)))) +
  ylab("% editing") +
  geom_blank(aes(y = ymin)) +
  geom_blank(aes(y = ymax))

## label for fig s13d
fig_s13d_label <- fig_s13bd_label +
  labs(title = "d")

# =============================================================================================

## filter donor screen for % HDR of total cutting
bfp_donor_screen_s13d2 <- bfp_donor_screen %>%
  filter(Editing == "HDR_Percent_Total" &
                                word(X_lab, start = 3) != "-")

## plot fig s13d, part 2 - % HDR of total cutting
fig_s13d2 <- ggplot(data = bfp_donor_screen_s13d2,
                   aes(X_lab, Percent)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21, 
                fill = pal[5], alpha = 0.7) +
  stat_summary(data = bfp_donor_screen_s13d2, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  scale_x_discrete(breaks = unique(bfp_donor_screen_s13d2$X_lab),
                   labels = addline_format(unique(bfp_donor_screen_s13d2$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 20), breaks = seq(0, 20, by = 5)) +
  ylab("% HDR of total editing")

# =============================================================================================

## filter for fold change in HDR
bfp_donor_screen_s13d3 <- bfp_donor_screen %>%
  filter(word(X_lab, start = 3) != "-" &
         Editing == "HDR") %>%
  mutate(sgRNA_group = word(X_lab, start = 2, end = 3)) %>%
  group_by(X_lab, Editing) %>%
  mutate(avg = mean(Percent)) %>%
  ungroup() %>%
  group_by(sgRNA_group, Editing) %>%
  mutate(fold_change = Percent / first(avg)) %>%
  ungroup()
  
## plot fig s13, panel 3 - fold change in HDR
fig_s13d3 <- ggplot(data = bfp_donor_screen_s13d3,
                    aes(X_lab, fold_change)) +
  geom_beeswarm(cex = 1.2, size = 2.5, pch = 21,
                fill = pal[3], alpha = 0.7) +
  stat_summary(data = bfp_donor_screen_s13d3, fun = "mean",
               geom = "point", pch = "_", size = 8) + 
  scale_x_discrete(breaks = unique(bfp_donor_screen_s13d3$X_lab),
                   labels = addline_format(unique(bfp_donor_screen_s13d3$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 4),
                     breaks = seq(0, 4, by = 1)) +
  ylab("Fold change in HDR")

## create source file for fig s13d
fig_s13d_source <- full_join(bfp_donor_screen_s13d1, bfp_donor_screen_s13d2) %>%
  left_join(bfp_donor_screen_s13d3) %>%
  mutate(Panel = "d") %>%
  select(Panel, Sample, X_lab, Editing, Percent, fold_change) %>%
  pivot_wider(names_from = Editing,
              values_from = c(Percent, fold_change)) %>%
  select(-c("fold_change_Indels", "fold_change_HDR_Percent_Total")) %>%
  rename_all(list(~str_replace_all(., "_", " "))) %>%
  rename_at(vars(contains("Percent ")), list(~gsub("Percent ", "", .))) %>%
  rename("X label" = `X lab`,
         "Fold change in HDR" = `fold change HDR`,
         "Percent HDR of total editing" = `HDR Total`)

## append to source list
source_data_list <- append(source_data_list, list("FigS13d" = fig_s13d_source))

# =============================================================================================

## plot fig s13a - flow gating for dRECS
fig_s13a <- ggdraw() +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  labs(title = "a") +
  draw_image(magick::image_read_pdf("Images/figs13a.pdf", density = 600))

## align bottom half of plot
fig_s13_bottom <- fig_s13b_label + fig_s13b1 + fig_s13bd_label + fig_s13b2 + fig_s13bd_label + fig_s13b3 +
  fig_s13c1_label + fig_s13c1 + fig_s13c_label + fig_s13c2 + fig_s13c_label + fig_s13c3 +
  fig_s13d_label + fig_s13d1 + fig_s13bd_label + fig_s13d2 + fig_s13bd_label + fig_s13d3 +
  plot_layout(ncol = 6, widths = c(0.15, 1))

## align with fig s13a
fig_s13 <- (fig_s13a + plot_spacer() + plot_layout(ncol = 2)) / fig_s13_bottom +
  plot_layout(heights = c(1, 3)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

## save fig s13
ggsave("Stat_analysis/Supp_figures/FigS13_dRECS.pdf", fig_s13,
       height = 24, width = 20)

# =============================================================================================

## select final experiment for fig 6
bfp_final <- bfp %>%
  filter(Experiment == 4 &
           Days == 4) %>%
  select(-Ratio_dRNA) %>%
  pivot_longer(cols = WT:HDR_Percent_Total,
               names_to = "Editing",
               values_to = "Percent") %>%
  ## alter X labels for figure plotting
  filter(X_lab == "+ - - -" |
         X_lab == "+ 1 - -" |
         X_lab == "+ 1 - 3" |
         X_lab == "+ 1 asym -" |
         X_lab == "+ 1 asym 3" |
         X_lab == "- - - -") %>%
  ## remove transfection control
  filter(!grepl("^turbo", Sample)) %>%
  ## make groups for facet plot in fig 6b
  mutate(facet_group = ifelse(Percent > 5, 1, 2),
         ymin = ifelse(facet_group == 1, 15, 0),
         ymax = ifelse(facet_group == 1, 35, 5),
         X_lab = gsub("asym", "+", X_lab),
         X_lab = gsub("[0-9]", "+", X_lab))

## subset for fig 6b
bfp_fig_6b <- bfp_final %>%
  filter(Editing == "HDR" | Editing == "Indels")

## subset for fig 6c
bfp_fig_6c <- bfp_final %>%
  filter(Editing == "HDR_Percent_Total" &
         X_lab != "+ - - -" &
         X_lab != "- - - -")

## plot fig 6b
fig_6b <- ggplot(data = bfp_fig_6b,
                 aes(factor(Order), Percent, fill = Editing)) +
  geom_beeswarm(cex = 2, size = 2.5, pch = 21, alpha = 0.7) +
  stat_summary(data = bfp_fig_6b, fun = "mean",
               geom = "point", pch = "_", size = 8) +
  facet_grid(facet_group ~ ., scales = "free") +
  theme(strip.background = element_blank(),
        strip.text.y = element_blank()) +
  scale_x_discrete(breaks = unique(bfp_fig_6b$Order),
                   labels = addline_format(unique(bfp_fig_6b$X_lab))) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = pal, breaks = unique(bfp_fig_6b$Editing),
                     labels = unique(gsub(".*\\s", "",
                                          unique(bfp_fig_6b$Editing)))) +
  ylab("% editing") +
  geom_blank(aes(y = ymin)) +
  geom_blank(aes(y = ymax)) +
  theme(panel.spacing = unit(15, "pt"))

## plot fig 6c
fig_6c <- ggplot(data = bfp_fig_6c, aes(factor(Order), Percent)) +
  geom_beeswarm(cex = 2, size = 2.5, pch = 21, alpha = 0.7, fill = pal[5]) +
  stat_summary(data = bfp_fig_6c, fun = "mean",
               geom = "point", pch = "_", size = 8) + 
  scale_x_discrete(breaks = unique(bfp_fig_6c$Order),
                   labels = addline_format(unique(bfp_fig_6c$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 25),
                     breaks = c(0, 5, 10, 15, 20, 25),
                     labels = c(0, 5, 10, 15, 20, 25)) +
  scale_fill_manual(values = pal[5], breaks = unique(bfp_fig_6c$Editing),
                     labels = unique(gsub(".*\\s", "", unique(bfp_fig_6c$Editing)))) +
  ylab("% HDR of total editing")

## create labels
fig_6_label <- label_plot(data = bfp_final,
                            aes(x = as_factor(Experiment),
                                y = Percent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\nCas9\n\nBFP sgRNA\n\nDonor\n\ndRNA")

fig_6b_label <- fig_6_label +
  labs(title = "b") 

fig_6c_label <- fig_6_label +
  labs(title = "c")

## plot fig 6a - schematic and alignment for dRECS
fig_6a <- ggdraw() +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  labs(title = "a") +
  draw_image(magick::image_read_pdf("Images/fig6a.pdf", density = 600))

## align bottom half plot
fig_6_bottom <- fig_6b_label + fig_6b + fig_6c_label + fig_6c +
  plot_layout(ncol = 4, widths = c(0.05, 1))

## align both plots
fig_6 <- (fig_6a / fig_6_bottom) +
  plot_layout(heights = c(2, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 20, l = 7, unit = "pt"))

## save fig 6 - dRECS
ggsave("Stat_analysis/Main_figures/Fig6_dRECS.pdf", fig_6, 
       height = 12, width = 12)

## create source file for fig 6b
fig_6b_source <- bfp_fig_6b %>%
  mutate(Panel = "b") %>%
  select(Panel, Sample, X_lab, Editing, Percent) %>%
  pivot_wider(names_from = Editing,
              values_from = Percent) %>%
  rename("X label" = X_lab)

## create source file for fig 6c
fig_6c_source <- bfp_fig_6c %>%
  mutate(Panel = "c") %>%
  select(Panel, Sample, X_lab, Percent) %>%
  rename("X label" = X_lab,
         "Percent HDR of total editing" = Percent)

## append to source list
source_data_list <- append(source_data_list, list("Fig6b" = fig_6b_source,
                                                  "Fig6c" = fig_6c_source))

```

```{r timepoints}

## import 72 hour data
timepoints <- read_csv("Data/Compiled_longer.csv") %>%
  filter(Reads_passing >= 1000,
         Time == 72,
         !grepl("VEGFA", Site)) %>%
  group_by(Site, X_lab, Time) %>%
  mutate(mean_indel = mean(Indel_stringent)) %>%
  ungroup() %>%
  mutate(on_ot = if_else(grepl("*ON", Site), "on-target", "off-target"),
         guide = gsub("\\s.*$", "", Site),
         on_ot = factor(on_ot, levels = c("on-target", "off-target")),
         X_lab = factor(X_lab, levels = c("+ + -", "+ + +", "- - -")),
         guide = gsub("g", " sgRNA", guide),
         Site = gsub("g", " sgRNA", Site),
         Experiment = as_factor(Experiment))

## create 72 hour timepoint plots for 3 sites
timepoint_plot <- timepoints %>%
  mutate(tag = letters[Experiment]) %>%
  group_by(Experiment) %>%
  nest() %>%
  mutate(plot1 = map2(data, Experiment, ~ggplot(data = .x) +
                       aes(X_lab, Indel_stringent, fill = on_ot) +
                       geom_beeswarm(cex = 1.2, size = 2.5, pch = 21, alpha = 0.7) +
                       stat_summary(data = .x, fun = "mean",
                                   geom = "point", pch = "_", size = 8) +
                       scale_x_discrete(labels = addline_format(unique(.$X_lab))) +
                       scale_y_continuous(expand = c(0, 0),
                                          limits = c(0, round_max(.)),
                                          breaks = seq(0, round_max(.),
                                                       by = if_else(round_max(.) <= 10, round_max(.) / 5,
                                                                    if_else(round_max(.) >= 30, 10, 5)))) + 
                       scale_fill_manual(values = pal) +
                       labs(y = "Indel frequency (%)",
                            title = unique(.$Site)[2])),
         label_plots1 = map2(data, Experiment,
                           ~label_plot(data = .x) +
                             aes(guide,
                                 Indel_stringent) +
                             geom_point(color = "NA") +
                             labs(title = unique(.$tag)) +
                             scale_x_discrete(labels = label_grab(.))))

## generate ratio plots as a list
ratio_timepoints <- timepoints %>%
  ratio_calc_longer() %>%
  group_by(Experiment) %>%
  nest() %>%
  mutate(plot2 = map2(data, Experiment, ~ggplot(data = .x) +
                       aes(X_lab, ratio) +
                       geom_jitter(size = 2.5, width = 0.05, pch = 21, alpha = 0.7,
                                   fill = pal[5]) +
                       stat_summary(data = .x, fun = "mean",
                                    geom = "point", pch = "_", size = 8) + # plots mean of each site
                       scale_x_discrete(labels = addline_format(unique(.$X_lab))) +
                       scale_fill_manual(values = pal[5]) +
                       scale_y_continuous(expand = c(0, 0),
                                          limits = c(0, round_max(.)),
                                          breaks = seq(0, round_max(.),
                                                       by = if_else(round_max(.) <= 10, round_max(.) / 5,
                                                                    if_else(round_max(.) >= 30, 50, 5)))) +
                       ylab("On-target/Off-target\nindel frequency ratio") +
                       ggtitle(paste0(unique(.$guide), " ", unique(na.omit(.$location_OT))))),
         label_plots2 = map2(data, Experiment,
                           ~label_plot(data = .x) +
                             aes(guide,
                                 ratio) +
                             geom_point(color = "NA") +
                             scale_x_discrete(labels = label_grab(.))))

## join indels and ratio plots
joined_timepoints <- inner_join(timepoint_plot, ratio_timepoints, by = "Experiment") %>%
  ## join to plot side by side
   pivot_longer(cols = c("label_plots1", "plot1", "label_plots2", "plot2"),
                names_to = "plottype",
                values_to = "plot")

## plot in order
fig_s7 <- wrap_plots(joined_timepoints$plot, ncol = 4, widths = c(0.15, 1, 0.05, 1)) &
  theme(plot.margin = margin(t = 7, r = 7, b = 25, l = 7, unit = "pt"))

## save fig s7 - 72 hour timepoints
ggsave("Stat_analysis/Supp_figures/FigS7_72h.pdf", fig_s7,
       height = 15, width = 9)

## create source file for fig s7
fig_s7_source <- timepoint_plot %>%
  select(Experiment, data) %>%
  unique() %>%
  unnest(data) %>%
  rename("Panel" = tag) %>%
  mutate(target = word(Site, -1),
         X_lab = factor(X_lab, levels = c("+ + -", "+ + +", "- - -"))) %>%
  indel_source() %>%
  arrange(Panel, `X label`) %>%
  mutate(Ratio = ifelse(`X label` == "- - -", NA, `Indels ON` / `Indels OT`))

## split by panel
fig_s7_list <- fig_s7_source %>%
  named_group_split(Panel)

## change names to figure panels
names(fig_s7_list) <- paste0("FigS7", names(fig_s7_list))

## append fig s5
source_data_list <- append(source_data_list, fig_s7_list)

```

```{r guide_seq}

## read in guide-seq validation data
gseq <- read_csv("Data/guideseq_validation.csv") %>%
  mutate(ODN_of_indel_ON = ODN_integration_ON / ON_indels * 100,
         ODN_of_indel_OT = ODN_integration_OT1 / OT_indels * 100,
         ODN_of_indel_OT = gsub(NaN, "0", ODN_of_indel_OT),
         ODN_of_indel_OT = as.numeric(ODN_of_indel_OT),
         Experiment = "U2OS") %>%
  pivot_longer(cols = ON_indels:ODN_of_indel_OT,
               names_to = "type",
               values_to = "percent") %>%
  filter(Sample != "GFP") %>%
  filter(Sample != "Electroporate_only")

## create label plot
fig_s2_label <- label_plot(data = gseq,
                           aes(x = Experiment,
                               y = percent)) +
  geom_point(color = "NA") +
  scale_x_discrete(labels = "\nCas9\n\nFANCF sgRNA2\n\ndRNA1\n\ndsODN")

## add tags to each label
fig_s2a_label <- fig_s2_label +
  labs(title = "a")

fig_s2b_label <- fig_s2_label +
  labs(title = "b")

fig_s2c_label <- fig_s2_label +
  labs(title = "c")

## plot indel frequency
fig_s2a <- ggplot(data = gseq %>% filter(grepl("*_indels", type)),
                    aes(fct_inorder(X_lab), percent, fill = type)) + 
  geom_point(size = 2.5, pch = 21, alpha = 0.7) +
  scale_x_discrete(labels = addline_format(unique(gseq$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 80), breaks = seq(0, 80, by = 20)) +
  scale_fill_manual(labels = c("on-target", "off-target"), values = pal) +
  labs(y = "Indel frequency (%)",
       title = "FANCF sgRNA2 OT1")

## plot ODN integration
fig_s2b <- ggplot(data = gseq %>% filter(grepl("ODN_integration*", type)),
                    aes(fct_inorder(X_lab), percent, fill = type)) + 
  geom_point(size = 2.5, pch = 21, alpha = 0.7) +
  scale_x_discrete(labels = addline_format(unique(gseq$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 20), breaks = seq(0, 20, by = 5)) +
  scale_fill_manual(labels = c("on-target", "off-target"), values = pal) +
  labs(y = "dsODN tag integration (%)",
       title = "FANCF sgRNA2 OT1")

## plot % ODN integration of indels detected
fig_s2c <- ggplot(data = gseq %>% filter(grepl("ODN_of_*", type)),
                    aes(fct_inorder(X_lab), percent, fill = type)) + 
  geom_point(size = 2.5, pch = 21, alpha = 0.7) +
  scale_x_discrete(labels = addline_format(unique(gseq$X_lab))) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 40), breaks = seq(0, 40, by = 10)) +
  scale_fill_manual(labels = c("on-target", "off-target"), values = pal) +
  labs(y = "dsODN integration /\n indel frequency (%)",
       title = "FANCF sgRNA2 OT1")

## layout all plots
fig_s2_top <- fig_s2a_label + plot_spacer() + fig_s2a +
  fig_s2b_label + plot_spacer() + fig_s2b +
  fig_s2c_label + plot_spacer() + fig_s2c +
  plot_layout(ncol = 9, widths = c(0.05, 0.05, 1))

## bring in guide-seq data pdf
fig_s2_bottom <- ggdraw() +
  theme_cowplot() + 
  theme(plot.title = element_text(size = 36, hjust = 0),
        axis.line = element_blank()) +
  labs(title = "d") +
  draw_image(magick::image_read_pdf("Images/figs2d.pdf", density = 600))

## plot together
fig_s2 <- fig_s2_top / fig_s2_bottom + plot_layout(heights = c(1, 3)) &
  theme(plot.margin = margin(t = 7, b = 30, r = 7, l = 7))

## save fig s2 - guide seq
ggsave("Stat_analysis/Supp_figures/FigS2_guide-seq.pdf", fig_s2,
       height = 16, width = 16)

## create source file for fig s2a
fig_s2a_source <- gseq %>%
  filter(grepl("*_indels", type)) %>%
  mutate(Guide = "FANCF sgRNA2",
         target = word(type, 1, sep = fixed("_")),
         Panel = "a") %>%
  select(Panel, everything()) %>%
  select(-type, -Experiment) %>%
  pivot_wider(names_from = target, 
              values_from = percent)  %>%
  rename("Name" = Sample,
         "Indels ON" = ON,
         "Indels OT" = OT)

## create source file for fig s2b
fig_s2b_source <- gseq %>%
  filter(grepl("ODN_integration*", type)) %>%
  mutate(Guide = "FANCF sgRNA2",
         Panel = "b") %>%
  select(Panel, everything()) %>%
  select(-Experiment) %>%
  pivot_wider(names_from = type, 
              values_from = percent) %>%
  rename_all(list(~str_replace_all(., "_", " ")))

## create source file for fig s2c
fig_s2c_source <- gseq %>%
  filter(grepl("ODN_of_*", type)) %>%
  mutate(Guide = "FANCF sgRNA2",
         Panel = "c") %>%
  select(Panel, everything()) %>%
  select(-Experiment) %>%
  pivot_wider(names_from = type, 
              values_from = percent) %>%
  rename("dsODN / indel frequency ON" = ODN_of_indel_ON,
         "dsODN / indel frequency OT1" = ODN_of_indel_OT)

## append to source data list
source_data_list <- append(source_data_list,
                           list("FigS2a" = fig_s2a_source,
                                "FigS2b" = fig_s2b_source,
                                "FigS2c" = fig_s2c_source))

## arrange and write all figures to .xlsx
source_data_list <- source_data_list[str_sort(names(source_data_list), numeric = TRUE)]
write.xlsx(source_data_list, file = "Stat_analysis/source_data_file.xlsx", append = TRUE)

## get all files used in analysis for data collection
dRNA_only_all <- read_csv("Data/dRNA_only_controls.csv") %>%
  filter(Experiment != 5) %>%
  mutate(source = "dRNA only")

titrations2 <- titrations %>%
  ungroup() %>%
  mutate(Experiment = as.numeric(Experiment)) %>%
  unnest(cols = c(Experiment, data)) %>%
  bind_rows(v3_titration) %>%
  mutate(X_lab = as.character(X_lab))

nontargeting2 <- nontargeting %>%
  ungroup() %>%
  mutate(Experiment = as.numeric(Experiment)) %>%
  unnest(cols = c(Experiment, data)) %>%
  bind_rows(nontargeting_FANCF)

timepoints2 <- timepoints %>%
  ungroup() %>%
  mutate(Experiment = as.numeric(Experiment)) %>%
  unnest(cols = as.numeric())

gseq2 <- read_csv("Data/guideseq_val_filenames.csv")

gseq_full <- read_csv("Data/guideseq_filenames.csv")

## bind together and classify
all_files <- bind_rows("dRNA screens" = dRNA_screens %>% filter(Experiment != 24),
                       "ciCas9" = cicas9,
                       "truncated VEGFA" = tru_vegfa,
                       "dRNA titrations" = titrations2,
                       "nontargeting" = nontargeting2,
                       "HiFi Cas9" = hifi,
                       "mdOTS" = mdots,
                       "dRNA only controls" = dRNA_only_all,
                       "U2OS" = u2os,
                       "ELF-1" = elf,
                       "72h timepoint" = timepoints2,
                       "GUIDEseq validation" = gseq2,
                       "GUIDEseq" = gseq_full, 
                       .id = "id") %>%
  filter(!is.na(Sample),
         !grepl("predicted", Site),
         !grepl("trG10", Site)) %>%
  mutate(filename = gsub("wtf-difflib_indels.csv", "", Sample),
         dRNA = case_when(str_detect(id, "mdOTS") ~ gsub("_", " ",
                                          gsub("GFP", "no dRNA",
                                               gsub("d", "-dRNA",
                                                    gsub("_[0-9]", "",
                                                         gsub("^[^g]*g[0-9]", "", Name))))),
           str_detect(Name, "d[0-9]s") ~ gsub("d", "dRNA", str_extract(Name, "d[0-9]s")),
           str_detect(Name, "d[0-9]") ~ gsub("d", "dRNA", str_extract(Name, "d[0-9]")),
           str_detect(Name, "GFP") ~ "no dRNA",
           str_detect(Name, "dNT") ~ "nontargeting dRNA",
           str_detect(Name, "NT") ~ "NT",
           str_detect(Name, "_0_[0-9]") ~ "no dRNA",
           str_detect(Name, "_[0-9]_[0-9]") ~ gsub("d", "dRNA", str_extract(Sample, "d[0-9]")),
           str_detect(Name, "_[0-9].[0-9]+_[0-9]") ~ gsub("d", "dRNA", str_extract(Sample, "d[0-9]")),
           str_detect(Name, "Electroporate") ~ "NT",
           id == "GUIDEseq" ~ "",
           str_detect(Name, "ODN") ~ gsub("_", " + ", Name),
           str_detect(Name, "FANCF sgRNA2 \\+ dRNA1") ~ "dRNA1",
           str_detect(Name, "dRNA1") ~ "dRNA1 only",
           str_detect(Name, "non-targeting") ~ "nontargeting dRNA",
           TRUE ~ "no dRNA"),
         library_layout = if_else(str_detect(filename, "assembled") == TRUE,
                                  "paired", "single"),
         instrument_model = if_else(id == "GUIDEseq", "Illumina MiSeq", "NextSeq 550"),
         library_ID = if_else(library_layout == "paired",
                              gsub("_pear.fastq.assembled.fastq", "", filename),
                              gsub(".fastq", "", filename)),
         replicate = case_when(str_detect(id, "GUIDE*") == TRUE ~ "1",
           str_detect(id, "ciCas9") == TRUE ~ as.character(Replicate),
           TRUE ~ word(Name, -1, sep = fixed("_"))),
         library_strategy = if_else(id == "GUIDEseq", "OTHER", "AMPLICON"),
         library_source = "GENOMIC",
         library_selection = "PCR",
         platform = "ILLUMINA",
         Site = case_when(str_detect(Site, "g[0-9]") == TRUE ~ gsub("g", " sgRNA", Site),
           id == "ciCas9" & Site == "on-target" ~ "FANCF sgRNA2 ON",
           id == "ciCas9" & Site == "off-target" ~ "FANCF sgRNA2 OT1",
           id == "GUIDEseq" ~ "whole genome",
           grepl("-R0[0-9]", Site) == TRUE ~ gsub("-R0[0-9]", "", Site),
           TRUE ~ Site),
         wording = paste(id, "- Genomic sequencing of the", Site),
         design_description = if_else(id == "GUIDEseq", "Whole genome off-target sequencing.", 
                                      "A locus of interest was amplified from genomic DNA and sequenced."),
         filetype = "fastq",
         title = case_when(id == "dRNA screens" & grepl("080917_NT_[0-9]_VEGFg3_ON", filename) == TRUE ~ paste0("HiFi Cas9 & ", wording, " locus. dRNA: ", dRNA, "; Cas9 species: none; Replicate: ", replicate),
                           id == "dRNA screens" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate:", replicate),
           id == "ciCas9" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Time:", Time, "hours; Replicate:", replicate),
           id == "truncated VEGFA" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate:", replicate),
           id == "dRNA titrations" & grepl("R03_GFP", filename) == TRUE ~ paste("nontargeting and", wording, " locus. dRNA: ", dRNA,
                                                                                "; dRNA:sgRNA ratio: ", X_lab,
                                                                                "; Replicate: ", replicate),
           id == "dRNA titrations" & grepl("R03_d4_1_1", filename) == TRUE ~ paste0("nontargeting and ", wording, " locus. dRNA: ", dRNA,
                                                                                "; dRNA:sgRNA ratio: ", X_lab,
                                                                                "; Replicate: ", replicate),
           id == "dRNA titrations" ~ paste0(wording, " locus. dRNA: ", dRNA, "; dRNA:sgRNA ratio: ", X_lab, "; Replicate: ", replicate),
           id == "nontargeting" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate:", replicate),
           id == "HiFi Cas9" & dRNA == "NT" & grepl("17115_NT_[0-9]_FANCFg2", library_ID) == TRUE ~ paste0("mdOTS, dRNA only controls, & ", wording, " locus. dRNA: ", dRNA, "; Cas9 species: none; Replicate: ", replicate),
           id == "HiFi Cas9" & dRNA == "NT" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Cas9 species: none; Replicate: ", replicate),
           id == "HiFi Cas9" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Cas9 species:", Cas9, "; Replicate: ", replicate),
           id == "mdOTS" & grepl("17115_NT_[0-9]_VEGFg2", filename) == TRUE ~ paste0("dRNA only controls & ", wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "mdOTS" & grepl("17115_NT_[0-9]_ZSCAN", filename) == TRUE ~ paste0("dRNA only controls & ", wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "mdOTS" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "dRNA only controls" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "U2OS" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "ELF-1" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "72h timepoint" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "GUIDEseq validation" ~ paste0(wording, " locus. dRNA: ", dRNA, "; Replicate: ", replicate),
           id == "GUIDEseq" ~ paste0(wording, ". ", File),
           TRUE ~ "fix")) %>%
  select(library_ID, title, library_strategy, library_source, library_selection, library_layout,
         platform, instrument_model, design_description, filetype, filename) %>%
  distinct(filename, .keep_all = TRUE) %>%
  #distinct() %>%
  mutate(filename3 = if_else(grepl("pear", filename),
                             gsub("pear.fastq.assembled.fastq", "R1_001.fastq", filename),
                             filename),
         filename2 = if_else(grepl("pear", filename),
                             gsub("pear.fastq.assembled.fastq", "R2_001.fastq", filename), "")) %>%
  select(-filename) %>%
  rename("filename" = filename3)

## write SRA metadata file
write_csv(all_files, "Seq_analysis/200426_SRA_file_descriptions.csv")

```
